<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="keywords" content="R, Bioconductor, Affymetrix">
  <meta name="author" content="Henrik Bengtsson">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="shortcut icon" type="image/png" href="../assets/ico/favicon.png">
  <title title="The Aroma Project (Aroma): Presentations">Aroma: Presentations</title>
  <link href="../assets/css/bootstrap.min.css" rel="stylesheet">
  <link href="../assets/css/cbc-tweaks.css" rel="stylesheet">
  <link href="../assets/css/r_highlight.css" rel="stylesheet">
  <script src="../assets/js/r_highlight.js"></script>
</head>
<body>
<a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>
<nav class="navbar navbar-default navbar-fixed-top" role="navigation" id="topnavbar">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html" title="The Aroma Project">Aroma</a>
    </div>

    <div class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../getstarted/index.html">Get Started</a></li>
        <li><a href="../docs/index.html">Documentation</a></li>
        <li><a href="../features/index.html">Features</a></li>
        <li><a href="../resources/index.html">Resources</a></li>
        <li><a href="../forum/index.html">Forum</a></li>
        <li><a href="../FAQ/index.html">FAQ</a></li>
        <li><a href="../developers/index.html">Developers' Corner</a></li>
        <li><a href="../about/index.html">About</a></li>
      </ul>
    </div>
  </div>
</nav>

<div id="wrap">
<div class="bs-docs-header" id="content">
<div class="container">
<h1>Presentations</h1>

<p>Below are selected presentations related to aroma-project.org and/or  related statistical methods by us.</p>

<h2>2012</h2>

<p>2012-12-14: Henrik Bengtsson, <em>aroma.seq: Bringing sequence analysis to the Aroma Framework</em>, <a href="http://www.fgcz.ch/Bioconductor2012">Bioconductor European Developers&#39; Workshop 2012</a>, UZH/ETH Zurich, 20+5 minutes.<br>
Slides (35): <a href="../share/presentations/BengtssonH_20121214-aroma.seq,BiocDevel2012.pdf">pdf</a> [0.6Mb]</p>

<h2>2011</h2>

<p>2011-10-27: Henrik Bengtsson, <em>Single Tumor-Normal Pair Parent-Specific Copy Number Analysis</em>, <a href="http://www.ipam.ucla.edu/programs/genws2/">Workshop II: Transcriptomics and Epigenomics</a>, IPAM, UCLA, Los Angeles, CA, 55 minutes.<br>
Slides (35+6): <a href="../share/presentations/BengtssonH_20111027-IPAM,full.pdf">pdf</a> [2.3Mb]</p>

<p>2011-07-28: Henrik Bengtsson, <em>Single Tumor-Normal Pair Parent-Specific Copy Number Analysis</em>, <a href="https://secure.bioconductor.org/BioC2011/">BioC2011</a>, Seattle, WA, 45 minutes.<br>
Slides (34): <a href="../share/presentations/BengtssonH_20110728-BioC2011.pdf">pdf</a> [2.1Mb]</p>

<p>2011-06-21: Henrik Bengtsson, <em>Single Tumor-Normal Pair Parent-Specific Copy Number Analysis</em>, Session I6 (IMS) on &#39;Statistical Methods for High-Throughput Genomic Assays&#39;, <a href="http://www.wnar.org/">WNAR 2011</a>, San Luis Obispo, CA, 25 minutes.<br>
Slides (28): <a href="../share/presentations/BengtssonH_20110621-WNAR2011.pdf">pdf</a> [1.7Mb]</p>

<p>2011-03-22: Henrik Bengtsson, <em>Enhanced power for segmenting parent-specific copy numbers</em>, Session #99 on &#39;Assessing DNA Copy Number Variation from High-Throughput Technologies&#39;, <a href="https://www.enar.org/meetings.cfm">ENAR 2011</a>, Miami, FL, 25 minutes.<br>
Slides (25): <a href="../share/presentations/BengtssonH_20110322-ENAR2011.pdf">pdf</a> [1.8Mb]</p>

<p>2011-03-10: Henrik Bengtsson, <em>Analysis of SNP arrays for parent-specific copy numbers</em>, Quantitative Biology Discussion Group (QBDG), Helen Diller Family Comprehensive Cancer Center, UCSF, 60 minutes.</p>

<h2>2010</h2>

<p>2010-11-10: Henrik Bengtsson, <em>Greatly improved parent-specific tumor copy numbers with DNA microarrays when a matched normal is available</em>, <a href="http://www.icgc.org/">International Cancer Genome Consortium</a> (ICGC), conference call, 60 minutes.</p>

<p>2010-09-30: Pierre Neuvial, <em>Statistical analysis of single nucleotide polymorphism microarrays in cancer studies</em>, <a href="http://med.stanford.edu/biostatistics/workshop.html">Workshop in Biostatistics</a>, Stanford University, CA.<br>
Slides (60): <a href="../share/presentations/NeuvialP_20100930-StanfordBiostatisticsWorkshop.pdf">pdf</a> [7.7Mb].</p>

<p>2010-08-18: Pierre Neuvial, <em>Greatly improved allele-specific tumor copy numbers with DNA microarrays when a matched normal is available</em>, CSB 2010 (International Conference on Computational Systems Bioinformatics), Stanford University, CA, USA.<br>
Slides (9+7): <a href="../share/presentations/neuvial,2010,CSB,TumorBoost,shortTalk.pdf">pdf</a> [1.0Mb].</p>

<p>2010-04-14: Pierre Neuvial, <em>On detecting and calling DNA copy number alterations in cancer samples from genotyping microarrays</em>, Institut Curie, Paris.<br>
Slides (46): <a href="../share/presentations/NeuvialP_2010spring-France.pdf">pdf</a> [2.1Mb].</p>

<h2>2009</h2>

<p>2009-10-01: Pierre Neuvial, <em>TumorBoost: Normalization of allele-specific tumor copy numbers in paired tumor/normal, designs for genotyping microarrays</em>, Statistics and Genomics Seminar, UC Berkeley, CA, USA.<br>
Slides (36): <a href="../share/presentations/neuvial_20091001_statGen.pdf">pdf</a> [2.1Mb].</p>

<p>2009-10-13: Henrik Bengtsson, <em>Improved copy number analysis</em>, Department of Epidemiology &amp; Biostatistics, University of California, San Francisco (UCSF), CA.<br>
Slides (46): <a href="../share/presentations/BengtssonH_20091013-Improved_copy_number_analysis.pdf">pdf</a> [4.6Mb].</p>

<p>2009-03-16: Henrik Bengtsson, <em>A Single-Array Preprocessing Method for Estimating Full-Resolution Raw Copy Numbers from all Affymetrix Genotyping Arrays</em>, Comprehending Copy Number Variation (Tools, Applications and Results), Cambridge Health Institute, March 16, 2009, San Diego, CA.<br>
 Slides (39): <a href="../share/presentations/BengtssonH_20090316-CNVMeeting,SanDiego.ppt">ppt</a> [0.8Mb], <a href="../share/presentations/BengtssonH_20090316-CNVMeeting,SanDiego.pdf">pdf</a> [0.8Mb]. Handouts: <a href="../share/presentations/BengtssonH_20090316-CNVMeeting,SanDiego,handouts,6up.pdf">6up.pdf</a>
[0.7Mb]</p>

<h2>2008</h2>

<p>2008-12-16: Henrik Bengtsson, <em>Low-Level Copy Number Analysis (CRMA v2 preprocessing)</em>, CEIT Workshop on SNP arrays, Dec 15-17, 2008, San Sebastian, Spain.<br>
 Slides (96): <a href="../share/presentations/BengtssonH_20081216-CEITCNWorkshop,Day2.ppt">ppt</a> [2.0Mb], <a href="../share/presentations/BengtssonH_20081216-CEITCNWorkshop,Day2.pdf">pdf</a> [2.1Mb]</p>

<p>2008-12-15: Henrik Bengtsson, <em>Low-Level Copy Number Analysis (Part 1-Background, Part 2-Simple preprocessing, Part 3-aroma.affymetrix)</em>, CEIT Workshop on SNP arrays, Dec 15-17, 2008, San Sebastian, Spain.<br>
 Slides (64): <a href="../share/presentations/BengtssonH_20081215-CEITCNWorkshop,Day1.ppt">ppt</a> [10.1Mb], <a href="../share/presentations/BengtssonH_20081215-CEITCNWorkshop,Day1.pdf">pdf</a> [6.6Mb]</p>

<p>2008-04-24: Henrik Bengtsson, <em>aroma.affymetrix - Processing very large Affymetrix data sets in bounded memory</em> (Working with Large Datasets), BioC 2008 developers meeting, Lausanne, Switzerland.<br>
 Slides: <a href="../share/presentations/BengtssonH_20080124-MBI_Ohio.ppt">ppt</a> [531kb], <a href="../share/presentations/BengtssonH_20080424-BioC2008-aroma.affymetrix.pdf">pdf</a> [445kb]</p>

<p>2008-01-24: Henrik Bengtsson, <em>Copy-number estimation on the latest generation of high-density oligonucleotide microarrays</em>, Mathematical Biosciences Institute, The Ohio State University.<br>
Slides: <a href="../share/presentations/BengtssonH_20080124-MBI_Ohio.ppt">ppt</a> [4.3Mb], <a href="../share/presentations/BengtssonH_20080124-MBI_Ohio.pdf">pdf</a> [1.9Mb]</p>

<h2>2007</h2>

<p>2007-09-13: Henrik Bengtsson, <em>Estimating chromosomal copy numbers from Affymetrix SNP &amp; CN chips</em>, Statistics and Genomics Seminar, UC Berkeley.<br>
 Slides: <a href="../share/presentations/BengtssonH_20070913-CRMA6.ppt">ppt</a> [3.3Mb], <a href="../share/presentations/BengtssonH_20070913-CRMA6.pdf">pdf</a> [1.7Mb]</p>

<p>2007-08-07: Henrik Bengtsson, <em>aroma.affymetrix - Analysis of small to very large Affymetrix data sets</em>, <a href="http://www.bioconductor.org/workshops/2007/BioC2007">BioC2007</a> lab session, 13:00-15:00, August 7, 2007, Fred Hutchinson Cancer Research Center, Seattle, US.<br>
 Slides: <a href="../share/presentations/BengtssonH_20070807-BioC2007lab.pdf">pdf</a> [0.5Mb], Suppl. slides: <a href="../share/presentations/BengtssonH_20070807-BioC2007sup.pdf">pdf</a> [0.9Mb], <a href="../share/presentations/BengtssonH_20070807-BioC2007sup.ppt">ppt</a> [3.4Mb]<br>
 Cut&#39;n&#39;paste code: <a href="../share/presentations/BengtssonH_20070807-BioC2007cutnpaste.R">src</a> [9kb]</p>

<p>2007-05-31: Henrik Bengtsson, <em>Copy-number estimation using Robust Multichip Analysis.</em><br>
 Thursday May 31, 2007 @ 9:30-10:45, INSERM Workshop #177 on Statistical Methods and novel strategies to search for genes involved in common diseases, Londe-Les-Maures (Toulon), France.<br>
 <a href="http://ist.inserm.fr/BASIS/elgis/fqmat/atelier/SDF?DOC_NUM=177">http://ist.inserm.fr/BASIS/elgis/fqmat/atelier/SDF?DOC_NUM=177</a></p>

<p>2007-04-21: Henrik Bengtsson, <em>Enhanced method for identifying copy-number aberrations in small to very large oligonucleotide microarray data sets.</em><br>
Saturday April 21, 2007 @ 13:30-14:00, Affymetrix Low Level Workshop, Oakland Marriott City Center, Oakland, 2007.<br>
<a href="http://www.affymetrix.com/corporate/events/low_level_workshop/2007_04_low_level_workshop.affx">http://www.affymetrix.com/corporate/events/low_level_workshop/2007_04_low_level_workshop.affx</a></p>

<p>2007-01-08: Henrik Bengtsson, <em>Crash course: Using aroma.affymetrix for copy-number analysis.</em><br>
Monday January 8, 2007 @ 13:30-16:00, Bioinformatics, WEHI, Melbourne.<br>
 <a href="http://www.maths.lth.se/bioinformatics/calendar/20070108b/">http://www.maths.lth.se/bioinformatics/calendar/20070108b/</a></p>

<h2>2006</h2>

<p>2006-09-28: Henrik Bengtsson, <em>Package aroma.affymetrix - How to analyze huge Affymetrix data sets in R on a notebook.</em><br>
 Thursday September 28, 2006 @ 16:00-17:00, Statistics and Genomics<br>
Seminar, 1011 Evans Hall, University of California, Berkeley<br>
 <a href="http://www.maths.lth.se/bioinformatics/calendar/20060928b/">http://www.maths.lth.se/bioinformatics/calendar/20060928b/</a></p>
</div>
</div> 
</div> 

<div id="filler"></div>


<div id="push"></div> 
<div id="footer" style="height: auto;">
  <div class="container">
<p class="muted credit" style="font-size: small;">
  The Aroma Project
  <span class="printonly">(http://www.aroma-project.org/)</span>
  ::
  &copy; Henrik Bengtsson
  ::
  Powered by <a href="http://cran.r-project.org/package=R.rsp">RSP</a>

  <span style="float: right;">
  <a href="http://validator.w3.org/check?st=1;outline=1;verbose=1;group=1&amp;uri=http://www.aroma-project.org/presentations/" aria-label="Validate HTML" id="validate-page" target="_blank"><span class="glyphicon glyphicon-check" title="Validate HTML" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Validate HTML</span></a>
  &nbsp;
  <a href="https://github.com/AromaProject/aroma.project.org-website/tree/master/content/presentations/index.md.rsp" id="edit"
  target="_blank"><span class="glyphicon glyphicon-edit" title="Edit page" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Edit page</span></a>
  </span>
</p>

  </div>
</div>

<div class="device-xs visible-xs"></div>
<div class="device-sm visible-sm"></div>
<div class="device-md visible-md"></div>
<div class="device-lg visible-lg"></div>

<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
<script src="../assets/js/post-jquery.js"></script>
<script src="https://netdna.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
<script src="../assets/js/post-bootstrap.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  
    ga('create', 'UA-11999401-1', 'auto');
    ga('send', 'pageview');
  </script>
</body>
</html>
