<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="keywords" content="R, Bioconductor, Affymetrix">
  <meta name="author" content="Henrik Bengtsson">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="shortcut icon" type="image/png" href="../assets/ico/favicon.png">
  <title title="The Aroma Project (Aroma): Add ons">Aroma: Add ons</title>
  <link href="../assets/css/bootstrap.min.css" rel="stylesheet">
  <link href="../assets/css/cbc-tweaks.css" rel="stylesheet">
  <link href="../assets/css/r_highlight.css" rel="stylesheet">
  <script src="../assets/js/r_highlight.js"></script>
</head>
<body>
<a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>
<nav class="navbar navbar-default navbar-fixed-top" role="navigation" id="topnavbar">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html" title="The Aroma Project">Aroma</a>
    </div>

    <div class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../getstarted/index.html">Get Started</a></li>
        <li><a href="../docs/index.html">Documentation</a></li>
        <li><a href="../features/index.html">Features</a></li>
        <li><a href="../resources/index.html">Resources</a></li>
        <li><a href="../forum/index.html">Forum</a></li>
        <li><a href="../FAQ/index.html">FAQ</a></li>
        <li><a href="../developers/index.html">Developers' Corner</a></li>
        <li><a href="../about/index.html">About</a></li>
      </ul>
    </div>
  </div>
</nav>

<div id="wrap">
<div class="bs-docs-header" id="content">
<div class="container">
<h1>Add ons</h1>

<h2>Addons to the Aroma Framework</h2>

<h3>Add on: ACNE</h3>

<p>Title: ACNE - Affymetrix SNP probe-summarization using non-negative
matrix factorization<br>
Authors: Maria Ortiz-Estevez, Angel Rubio, Henrik Bengtsson<br>
Description: Package for NMF summarization of SNP probes.<br>
Chip types: All Affymetrix genotyping and CN arrays.<br>
Requirements: Cross platform, aroma.affymetrix<br>
References: <a href="../publications/index.html">M. Ortiz-Estevez, H. Bengtsson, A. Rubio, ACNE: a
summarization method to estimate allele-specific copy numbers for
Affymetrix SNP arrays, Bioinformatics, 2010</a>.<br>
URL: <a href="https://r-forge.r-project.org/R/?group_id=587">https://r-forge.r-project.org/R/?group_id=587</a></p>

<h3>Add on: CalMaTe</h3>

<p>Title: CalMaTe - A post-calibration process to improve allele-specific
copy number estimates from SNP microarrays<br>
Authors: Maria Ortiz-Estevez, Ander Aramburu, Henrik Bengtsson, Pierre
Neuvial, Angel Rubio.<br>
Description: Package for calibrating allele-specific copy number
estimates (ASCNs).<br>
Chip types: Any genotyping and CN arrays, e.g. Affymetrix and Illumina.<br>
Requirements: Cross platform, aroma.core<br>
References: <a href="../publications/index.html">Ortiz-Estevez et al., CalMaTe: A method and software to
improve allele-specific copy number of SNP arrays for downstream
segmentation, Bioinformatics, 2012</a>.<br>
URL: <a href="https://r-forge.r-project.org/R/?group_id=587">https://r-forge.r-project.org/R/?group_id=587</a></p>

<h3>Add on: FIRMAGene</h3>

<p>Title: FIRMAGene - differential splicing for Affymetrix Gene 1.0 ST<br>
 Authors: Mark Robinson<br>
 Description:<br>
 Chip types: Affymetrix Gene 1.0 ST<br>
 Requirements: Cross platform, aroma.affymetrix<br>
 References: <a href="../publications/index.html">M. Robinson and TS. Speed, Differential splicing using
whole-transcript microarrays, BMC
Bioinformatics, 2009</a>.<br>
 URL: <a href="https://r-forge.r-project.org/R/?group_id=424">https://r-forge.r-project.org/R/?group_id=424</a><br>
URL: <a href="http://bioinf.wehi.edu.au/folders/firmagene/">http://bioinf.wehi.edu.au/folders/firmagene/</a></p>

<h3>Add on: GADA (Genome Alteration Detection Analysis)</h3>

<p>Authors: Pique-Regi et al.<br>
 Description: &quot;GADA is a fast and accurate method for detecting copy
number alterations (CNA) from array data. [...]&quot;<br>
 Chip types: Generic segmentation method applicable to all raw CNs.<br>
 Operating system: ?<br>
 References: (1) Pique-Regi R and Gonzalez JR: &quot;R-Gada: a package for
fast and parallel detection of copy number on multiple samples and
visualization&quot;,  <em>Bioinformatics</em>, Submitted Dec 2008, (2) Pique-Regi R,
Monso-Varona J,Ortega A, Seeger RC, Triche TJ, Asgharzadeh S: &quot;Sparse
representation and Bayesian detection of the genome copy number
alterations from microarray data&quot;, <em>Bioinformatics</em>, Feb 2008<br>
URL: <a href="https://r-forge.r-project.org/R/?group_id=915">https://r-forge.r-project.org/R/?group_id=915</a><br>
URL: <a href="https://groups.google.com//group/gadaproject">https://groups.google.com//group/gadaproject</a><br>
URL: <a href="http://biron.usc.edu/%7Epiquereg/GADA/">http://biron.usc.edu/~piquereg/GADA/</a></p>

<h2>Other tools leveraging aroma.* packages</h2>

<h3>AMDA</h3>

<p>Authors: Genopolis<br>
Operating system: Cross platform<br>
Description: Automated Microarray Data Analysis (AMDA) software<br>
References:<br>
URL: <a href="http://www.genopolis.it/">http://www.genopolis.it/</a></p>

<h3>BRB-ArrayTools</h3>

<p>Authors: Richard Simon &amp; BRB-ArrayTools Development Team<br>
Operating system: Windows with MS Excel<br>
Description: &quot;BRB-ArrayTools is an integrated package for the
visualization and statistical analysis of DNA microarray gene expression
data. It was developed by professional statisticians experienced in the
analysis of microarray data and involved in the development of improved
methods for the design and analysis of microarray based experiments. The
array tools package utilizes an Excel front end. Scientists are familiar
with Excel and utilizing Excel as the front end makes the system
portable and not tied to any database. [...]&quot;<br>
References:<br>
URL: <a href="https://linus./nci.nih.gov/BRB-ArrayTools.html">https://linus./nci.nih.gov/BRB-ArrayTools.html</a></p>

<h3>Expander</h3>

<p>Authors: Ron Shamir&#39;s Computational Genomics Laboratory, School of
Computer Science, Tel Aviv University<br>
Operating system: Java<br>
Description: &quot;EXPANDER (EXpression Analyzer and DisplayER) is a
java-based tool for analysis of gene expression data. It is capable of
(1) clustering (2) visualizing (3) biclustering and (4) performing
downstream analysis of clusters and biclusters such as functional
enrichment and promoter analysis. In particular, it can analyze groups
of genes for enrichment of transcription factor binding sites in their
promoters.&quot;<br>
References:<br>
URL: <a href="http://acgt.cs.tau.ac.il/expander/">http://acgt.cs.tau.ac.il/expander/</a></p>

<h3>O-miner</h3>

<p>Authors: Cutts et al, Molecular Oncology, Barts Cancer Institute, Queen
Mary&#39;s School of Medicine and Dentistry, UK.<br>
Operating system: Online Server<br>
Description: &quot;O-miner is a project designed to automate the analysis and
annotation of complex -omics data. The software is easy to use and
freely available to the scientific community. Users can upload data via
a web page and can be notified by email when their analysis is
completed. The output includes detailed data analysis reports and
diagrams.  [As of May 2012] genomics and transcriptomics analysis
options are available&quot;.<br>
References: RJ Cutts et al. O-miner: an integrative platform for
automated analysis and mining of -omics data, NAR 2012.<br>
URL: <a href="http://o-miner.org/">http://o-miner.org/</a></p>

<h3>PECA (Probe-level expression change averaging)</h3>

<p>Authors: Laura L. Elo,  Jukka Hiissa<br>
 Description: The PECA R-package implements the Probe-level Expression
Change Averaging (PECA) procedure for the identification of
differentially expressed genes or differentially spliced exons in
Affymetrix gene or exon microarray studies.<br>
References:<br>
URL: <a href="http://www.utu.fi/en/units/sci/units/math/Research/biomathematics/projects/Pages/peca.aspx">http://www.utu.fi/en/units/sci/units/math/Research/biomathematics/projects/Pages/peca.aspx</a></p>

<h3>Progenetix / tastyAroma</h3>

<p>Authors: M Baudis<br>
 Description: &quot;tastyAroma is a project for providing interfaces and
standard procedures as well as advanced plotting functions to the
aroma.affymetrix array analysis packages. ... Additionally, the
tastyAroma project provides scripts for the integration of analysis
results into a web portal, relying on libraries of the Progenetix
project.<br>
References:<br>
URL: <a href="https://www.progenetix.net//">https://www.progenetix.net//</a><br>
URL: <a href="https://www.progenetix.net//progenetixWiki/pmwiki.php/TastyAroma/TastyAroma">https://www.progenetix.net//progenetixWiki/pmwiki.php/TastyAroma/TastyAroma</a></p>

<h3>Repitools</h3>

<p>Authors: A. Statham et al.<br>
Description: &quot;Useful functions for quality assessment, visualization,
summarization and statistical analysis of epigenomics experiments.&quot;<br>
References: A. Statham et al., Repitools: an R package for the analysis
of enrichment-based epigenomic data, Bioinformatics, 2010.<br>
URL: <a href="http://repitools.r-forge.r-project.org/">http://repitools.r-forge.r-project.org/</a></p>
</div>
</div> 
</div> 

<div id="filler"></div>


<div id="push"></div> 
<div id="footer" style="height: auto;">
  <div class="container">
<p class="muted credit" style="font-size: small;">
  The Aroma Project
  <span class="printonly">(https://www.aroma-project.org/)</span>
  ::
  &copy; Henrik Bengtsson
  ::
  Powered by <a href="https://cran.r-project.org/package=R.rsp">RSP</a>

  <span style="float: right;">
  <a href="https://validator.w3.org/check?st=1;outline=1;verbose=1;group=1&amp;uri=https://www.aroma-project.org/addons/" aria-label="Validate HTML" id="validate-page" target="_blank"><span class="glyphicon glyphicon-check" title="Validate HTML" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Validate HTML</span></a>
  &nbsp;
  <a href="https://github.com/AromaProject/aroma.project.org-website/tree/master/content/addons/index.md.rsp" id="edit"
  target="_blank"><span class="glyphicon glyphicon-edit" title="Edit page" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Edit page</span></a>
  </span>
</p>

  </div>
</div>

<div class="device-xs visible-xs"></div>
<div class="device-sm visible-sm"></div>
<div class="device-md visible-md"></div>
<div class="device-lg visible-lg"></div>

<script src="../assets/js/jquery.min.js"></script>
<script src="../assets/js/post-jquery.js"></script>
<script src="../assets/js/bootstrap.min.js"></script>
<script src="../assets/js/post-bootstrap.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  
    ga('create', 'UA-11999401-1', 'auto');
    ga('send', 'pageview');
  </script>
</body>
</html>
