<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="keywords" content="R, Bioconductor, Affymetrix">
  <meta name="author" content="Henrik Bengtsson">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="shortcut icon" type="image/png" href="../../assets/ico/favicon.png">
  <title title="The Aroma Project (Aroma): Vignette: Naive genotype calls and confidence scores">Aroma: Vignette: Naive genotype calls and confidence scores</title>
  <link href="../../assets/css/bootstrap.min.css" rel="stylesheet">
  <link href="../../assets/css/cbc-tweaks.css" rel="stylesheet">
  <link href="../../assets/css/r_highlight.css" rel="stylesheet">
  <script src="../../assets/js/r_highlight.js"></script>
</head>
<body>
<a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>
<nav class="navbar navbar-default navbar-fixed-top" role="navigation" id="topnavbar">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../../index.html" title="The Aroma Project">Aroma</a>
    </div>

    <div class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../../getstarted/index.html">Get Started</a></li>
        <li><a href="../../docs/index.html">Documentation</a></li>
        <li><a href="../../features/index.html">Features</a></li>
        <li><a href="../../resources/index.html">Resources</a></li>
        <li><a href="../../forum/index.html">Forum</a></li>
        <li><a href="../../FAQ/index.html">FAQ</a></li>
        <li><a href="../../developers/index.html">Developers' Corner</a></li>
        <li><a href="../../about/index.html">About</a></li>
      </ul>
    </div>
  </div>
</nav>

<div id="wrap">
<div class="bs-docs-header" id="content">
<div class="container">
<h1>Vignette: Naive genotype calls and confidence scores</h1>

<h2>Vignette: Naive genotype calls and confidence scores</h2>

<p>Author: Pierre Neuvial<br>
 Created: 2010-03-04<br>
 Last updated: 2011-03-07</p>

<p>The TumorBoost method for normalizing allele-specific copy number based
on a pair of tumor-normal genotyping microarrays uses genotypes of <em>the
normal sample</em> for normalizing allele B fractions in a tumor. Here we
show how such genotypes can be obtained from allele B fractions in a
normal sample.  This is called &quot;naive genotyping&quot; because genotypes are
simply called based on local minima of an estimate of the density of
allele fractions in the normal sample, as described in
<a name=cite-BengtssonH_etal_2010-TumorBoost></a><a href="https://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;amp;cmd=prlinks&amp;amp;retmode=ref&amp;amp;id=20462408">Bengtsson, Neuvial, and Speed (2010)</a>.  We also illustrate how
associated genotype confidence score files can be created.</p>

<p><em>WARNING: This naive genotyper is intended for identifying heterozygous
SNPs to be used by TumorBoost and/or parent-specific CN segmentation
methods, which are methods that are very forgiving on genotyping errors
scattered along the genome.  It was not designed to be a per-SNP
high-performing genotyper.</em></p>

<h3>Data</h3>

<pre><code>totalAndFracBData/
  TCGA,GBM,BeadStudio,XY/
    HumanHap550/
      TCGA-02-0001-01C-01D-0184-06,fracB.asb
      TCGA-02-0001-01C-01D-0184-06,total.asb
      **TCGA-02-0001-10A-01D-0184-06,fracB.asb**
      TCGA-02-0001-10A-01D-0184-06,total.asb

  TCGA,GBM,CRMAv2/
    GenomeWideSNP/
      TCGA-02-0001-01C-01D-0182-01,fracB.asb
      TCGA-02-0001-01C-01D-0182-01,total.asb
      **TCGA-02-0001-10A-01D-0182-01,fracB.asb**
      TCGA-02-0001-10A-01D-0182-01,total.asb

annotationData/
  chipTypes/
    HumanHap550/
      HumanHap550,TCGA,HB20080512.ugp
      HumanHap550,TCGA,HB20100107,unitNames.txt  

    GenomeWideSNP_6/
      GenomeWideSNP_6,Full.CDF
      GenomeWideSNP_6,Full,na26,HB20080821.ugp
</code></pre>

<p>Note: above we are showing the same data sets as in the <a href="../../vignettes/tumorboost-highlevel/index.html">TumorBoost
vignette</a>, but in the present
vignette <em>only data files corresponding to allelic ratios in normal
samples</em> are used: here, &quot;TCGA-<em>-10A-</em>,fracB.asb&quot;, for normal blood.</p>

<h3>Setup</h3>

<pre><code class="r">library(&quot;aroma.cn&quot;)
log &lt;- verbose &lt;- Arguments$getVerbose(-8, timestamp=TRUE)
rootPath &lt;- &quot;totalAndFracBData&quot;
rootPath &lt;- Arguments$getReadablePath(rootPath)

dataSets &lt;- c(&quot;TCGA,GBM,BeadStudio,XY&quot;, &quot;TCGA,GBM,CRMAv2&quot;)
dataSet &lt;- dataSets[1]  ## Work with Illumina data

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Load the raw (tumor,normal) data set
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
ds &lt;- AromaUnitFracBCnBinarySet$byName(dataSet, chipType=&quot;*&quot;, paths=rootPath)
setFullNamesTranslator(ds, function(names, ...) {
  pattern &lt;- &quot;^(TCGA-[0-9]{2}-[0-9]{4})-([0-9]{2}[A-Z])[-]*(.*)&quot;
  gsub(pattern, &quot;\\1,\\2,\\3&quot;, names)
})
print(ds)
</code></pre>

<p>This gives</p>

<pre><code>AromaUnitFracBCnBinarySet:
Name: TCGA
Tags: GBM,BeadStudio,XY
Full name: TCGA,GBM,BeadStudio,XY
Number of files: 2
Names: TCGA-02-0001, TCGA-02-0001 [2]
Path (to the first file): totalAndFracBData/TCGA,GBM,BeadStudio,XY/HumanHap550
Total file size: 4.28 MB
RAM: 0.00MB
</code></pre>

<h3>Extract the normals</h3>

<pre><code class="r">types &lt;- sapply(ds, FUN=function(df) getTags(df)[1])
normals &lt;- grep(&quot;(10|11)[A-Z]&quot;, types)
dsN &lt;- ds[normals]
print(dsN)
</code></pre>

<p>This gives</p>

<pre><code>AromaUnitFracBCnBinarySet:
Name: TCGA
Tags: GBM,BeadStudio,XY
Full name: TCGA,GBM,BeadStudio,XY
Number of files: 1
Names: TCGA-02-0001 [1]
Path (to the first file): totalAndFracBData/TCGA,GBM,BeadStudio,XY/HumanHap550
Total file size: 2.14 MB
RAM: 0.00MB
</code></pre>

<h3>Naive genotype calling and associated confidence scores</h3>

<pre><code class="r">fullname &lt;- paste(c(getFullName(dsN), &quot;NGC&quot;), collapse=&quot;,&quot;)
chipType &lt;- getChipType(dsN, fullname=FALSE)
outPath &lt;- file.path(&quot;callData&quot;, fullname, chipType)

units &lt;- NULL
if (is.null(units)) {
  df &lt;- dsN[[1]]
  units &lt;- seq(length=nbrOfUnits(df))
}

adjust &lt;- 1.5

# Identify units on ChrX and ChrY
ugp &lt;- getAromaUgpFile(dsN)
units23 &lt;- getUnitsOnChromosome(ugp, 23)
is23 &lt;- is.element(units, units23)
units24 &lt;- getUnitsOnChromosome(ugp, 24)
is24 &lt;- is.element(units, units24)

kk &lt;- 1
dfN &lt;- dsN[[kk]]

tags &lt;- getTags(dfN)
tags &lt;- setdiff(tags, &quot;fracB&quot;)
tags &lt;- c(tags, &quot;genotypes&quot;)
fullname &lt;- paste(c(getName(dfN), tags), collapse=&quot;,&quot;)

filename &lt;- sprintf(&quot;%s.acf&quot;, fullname)
gcPathname &lt;- Arguments$getWritablePathname(filename, path=outPath, mustNotExist=FALSE)

csTags &lt;- c(tags, &quot;confidenceScores&quot;)
fullname &lt;- paste(c(getName(dfN), csTags), collapse=&quot;,&quot;)
filename &lt;- sprintf(&quot;%s.acf&quot;, fullname)
csPathname &lt;- Arguments$getWritablePathname(filename, path=outPath, mustNotExist=FALSE)

if (isFile(gcPathname) &amp;&amp; isFile(csPathname)) {
  next
}

betaN &lt;- dfN[units,1,drop=TRUE]

# Call gender
gender &lt;- callXXorXY(betaN[is23], betaN[is24], adjust=adjust, from=0, to=1)

# Call genotypes
naValue &lt;- as.double(NA)
fit &lt;- NULL
mu &lt;- rep(naValue, times=length(units))
cs &lt;- rep(naValue, times=length(units))

if (gender == &quot;XY&quot;) {
  # All but ChrX &amp; ChrY in male
  isDiploid &lt;- (!(is23 | is24))
  use &lt;- which(isDiploid)
  muT &lt;- callNaiveGenotypes(betaN[use], cn=2, adjust=adjust, from=0, to=1,
                                        verbose=less(verbose,10))
  fit &lt;- attr(muT, &#39;modelFit&#39;)
  mu[use] &lt;- muT
  use &lt;- which(!isDiploid)
  muT &lt;- callNaiveGenotypes(betaN[use], cn=1, adjust=adjust, from=0, to=1,
                                         verbose=less(verbose,10))
  mu[use] &lt;- muT
} else {
  # All but ChrY in female
  isDiploid &lt;- (!is24)
  use &lt;- which(isDiploid)
  muT &lt;- callNaiveGenotypes(betaN[use], cn=2, adjust=adjust, from=0, to=1,
                                        verbose=less(verbose,10))
  fit &lt;- attr(muT, &#39;modelFit&#39;)
  mu[use] &lt;- muT
}
print(table(mu, exclude=NULL))

# Translate genotype calls in fracB space to (AA,AB,BB,...)
calls &lt;- rep(as.character(NA), times=length(mu))
calls[mu ==   0] &lt;- &quot;AA&quot;
calls[mu == 1/2] &lt;- &quot;AB&quot;
calls[mu ==   1] &lt;- &quot;BB&quot;
print(table(calls, exclude=NULL))

# Calculate confidence scores
a &lt;- fit[[1]]$fitValleys$x[1]
b &lt;- fit[[1]]$fitValleys$x[2]
cs[isDiploid] &lt;- rowMins(abs(cbind(betaN[isDiploid]-a, betaN[isDiploid]-b)))
print(table(mu, exclude=NULL))

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Writing genotype calls (via temporary file)
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
pathname &lt;- gcPathname
pathnameT &lt;- sprintf(&quot;%s.tmp&quot;, pathname)
nbrOfUnits &lt;- nbrOfUnits(dfN)
gfN &lt;- AromaUnitGenotypeCallFile$allocate(pathnameT, platform=getPlatform(dfN), chipType=getChipType(dfN), nbrOfRows=nbrOfUnits)
footer &lt;- readFooter(gfN)
footer$method &lt;- &quot;NaiveGenotypeCaller&quot;
writeFooter(gfN, footer)

updateGenotypes(gfN, units=units, calls=calls)

res &lt;- file.rename(pathnameT, pathname)
if (!isFile(pathname)) {
  throw(&quot;Failed to rename temporary file: &quot;, pathnameT, &quot; -&gt; &quot;, pathname)
}
if (isFile(pathnameT)) {
  throw(&quot;Failed to rename temporary file: &quot;, pathnameT, &quot; -&gt; &quot;, pathname)
}

gfN &lt;- AromaUnitGenotypeCallFile(pathname)

print(gfN)




# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Writing confidence scores (via temporary file)
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
pathname &lt;- csPathname
pathnameT &lt;- sprintf(&quot;%s.tmp&quot;, pathname)
nbrOfUnits &lt;- nbrOfUnits(dfN)
csfN &lt;- AromaUnitSignalBinaryFile$allocate(pathnameT, platform=getPlatform(dfN), chipType=getChipType(dfN), nbrOfRows=nbrOfUnits, types=&quot;double&quot;, size=4, signed=TRUE)
footer &lt;- readFooter(csfN)
footer$method &lt;- &quot;NaiveGenotypeConfidenceScoreEstimator&quot;
writeFooter(csfN, footer)

csfN[units, 1] &lt;- cs

res &lt;- file.rename(pathnameT, pathname)
if (!isFile(pathname)) {
  throw(&quot;Failed to rename temporary file: &quot;, pathnameT, &quot; -&gt; &quot;, pathname)
}
if (isFile(pathnameT)) {
  throw(&quot;Failed to rename temporary file: &quot;, pathnameT, &quot; -&gt; &quot;, pathname)
}

cfN &lt;- AromaUnitSignalBinaryFile(pathname)
print(cfN)
</code></pre>

<p>The data sets created are set up as follows:</p>

<pre><code>callData/
  TCGA,GBM,BeadStudio,XY,NGC/
    HumanHap550/
      TCGA-02-0001,10A,01D-0184-06,genotypes,confidenceScores.acf
      TCGA-02-0001,10A,01D-0184-06,genotypes.acf
</code></pre>

<p>They can be loaded by:</p>

<pre><code class="r">gcN &lt;- AromaUnitGenotypeCallSet$byName(dataSet, tags=&quot;NGC&quot;, chipType=&quot;*&quot;)
print(gcN)
</code></pre>

<p>This gives:</p>

<pre><code>AromaUnitGenotypeCallSet:Name: TCGA
Tags: GBM,BeadStudio,XY,NGC
Full name: TCGA,GBM,BeadStudio,XY,NGC
Number of files: 1
Names: TCGA-02-0001 [1]
Path (to the first file): callData/TCGA,GBM,BeadStudio,XY,NGC/HumanHap550
Total file size: 1.07 MB
RAM: 0.00MB
</code></pre>

<pre><code class="r">csN &lt;- AromaUnitSignalBinarySet$byName(dataSet, tags=&quot;NGC&quot;, chipType=&quot;*&quot;, pattern=&quot;confidenceScores&quot;, paths=&quot;callData&quot;)
print(csN)
</code></pre>

<p>This gives:</p>

<pre><code>AromaUnitSignalBinarySet:
Name: TCGA
Tags: GBM,BeadStudio,XY,NGC
Full name: TCGA,GBM,BeadStudio,XY,NGC
Number of files: 1
Names: TCGA-02-0001 [1]
Path (to the first file): callData/TCGA,GBM,BeadStudio,XY,NGC/HumanHap550
Total file size: 2.14 MB
RAM: 0.00MB
</code></pre>

<h2>References</h2>

<p><a
name=bib-BengtssonH_etal_2010-TumorBoost></a><a href="#cite-BengtssonH_etal_2010-TumorBoost">[1]</a>
H. Bengtsson, P. Neuvial and T. P. Speed. &quot;TumorBoost: normalization of
allele-specific tumor copy numbers from a single pair of tumor-normal
genotyping microarrays&quot;. Eng. In: <em>BMC bioinformatics</em> 11 (2010), p.
245. ISSN: 1471-2105. DOI:
<a href="http://dx.doi.org/10.1186/1471-2105-11-245">10.1186/1471-2105-11-245</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=20462408">20462408</a>.</p>
</div>
</div> 
</div> 

<div id="filler"></div>


<div id="push"></div> 
<div id="footer" style="height: auto;">
  <div class="container">
<p class="muted credit" style="font-size: small;">
  The Aroma Project
  <span class="printonly">(http://www.aroma-project.org/)</span>
  ::
  &copy; Henrik Bengtsson
  ::
  Powered by <a href="http://cran.r-project.org/package=R.rsp">RSP</a>

  <span style="float: right;">
  <a href="http://validator.w3.org/check?st=1;outline=1;verbose=1;group=1&amp;uri=http://www.aroma-project.org/vignettes/naive-genotyping/" aria-label="Validate HTML" id="validate-page" target="_blank"><span class="glyphicon glyphicon-check" title="Validate HTML" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Validate HTML</span></a>
  &nbsp;
  <a href="https://github.com/AromaProject/aroma.project.org-website/tree/master/content/vignettes/naive-genotyping/index.md.rsp" id="edit"
  target="_blank"><span class="glyphicon glyphicon-edit" title="Edit page" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Edit page</span></a>
  &nbsp;
  <a href="http://v1.aroma-project.org/vignettes/naive-genotyping/" id="old-page"
  target="_blank"><span class="glyphicon glyphicon-time" title="Go to old page" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Go to old page</span></a>
  </span>
</p>

  </div>
</div>

<div class="device-xs visible-xs"></div>
<div class="device-sm visible-sm"></div>
<div class="device-md visible-md"></div>
<div class="device-lg visible-lg"></div>

<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
<script src="../../assets/js/post-jquery.js"></script>
<script src="http://netdna.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
<script src="../../assets/js/post-bootstrap.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  
    ga('create', 'UA-11999401-1', 'auto');
    ga('send', 'pageview');
  </script>
</body>
</html>
