<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="keywords" content="R, Bioconductor, Affymetrix">
  <meta name="author" content="Henrik Bengtsson">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="shortcut icon" type="image/png" href="../../assets/ico/favicon.png">
  <title title="The Aroma Project (Aroma): Resource: Affymetrix microarray data sets">Aroma: Resource: Affymetrix microarray data sets</title>
  <link href="../../assets/css/bootstrap.min.css" rel="stylesheet">
  <link href="../../assets/css/cbc-tweaks.css" rel="stylesheet">
  <link href="../../assets/css/r_highlight.css" rel="stylesheet">
  <script src="../../assets/js/r_highlight.js"></script>
</head>
<body>
<a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>
<nav class="navbar navbar-default navbar-fixed-top" role="navigation" id="topnavbar">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../../index.html" title="The Aroma Project">Aroma</a>
    </div>

    <div class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../../getstarted/index.html">Get Started</a></li>
        <li><a href="../../docs/index.html">Documentation</a></li>
        <li><a href="../../features/index.html">Features</a></li>
        <li><a href="../../resources/index.html">Resources</a></li>
        <li><a href="../../forum/index.html">Forum</a></li>
        <li><a href="../../FAQ/index.html">FAQ</a></li>
        <li><a href="../../developers/index.html">Developers' Corner</a></li>
        <li><a href="../../about/index.html">About</a></li>
      </ul>
    </div>
  </div>
</nav>

<div id="wrap">
<div class="bs-docs-header" id="content">
<div class="container">
<h1>Resource: Affymetrix microarray data sets</h1>

<h2>Data sets from Affymetrix Inc.</h2>

<p><strong>Affymetrix: Genome-Wide Human SNP Array 5.0 Sample Data Set</strong><br>
Chip type(s): GenomeWideSNP_5<br>
Description: 30 (10 trios) HapMap CEU samples.  Among these, 4 are
replicated 4 times, and one is replicated 3 times.  In total
25+4*4+1*3 = 44 hybridizations.<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/genomewide_snp5_data.affx">http://www.affymetrix.com/support/technical/sample_data/genomewide_snp5_data.affx</a></p>

<p><strong>Affymetrix: Genome-Wide Human SNP Array 6.0 Sample Data Set</strong><br>
Chip type(s): GenomeWideSNP_6<br>
Description: 270 HapMap samples data set [270 hybridizations], and
Chromosome X Titration data set (1X, 2X, 3X, 4X, 5X) of five replicates
[5x5=25 hybridizations].  Note: According to some people, some 5X and 4X
have been &quot;swapped&quot;,
cf. <a href="http://dchip.forum5.com/viewtopic.php?t=179&amp;mforum=dchip">http://dchip.forum5.com/viewtopic.php?t=179&amp;mforum=dchip</a>.<br>
Note: This data set need to be ordered from Affymetrix and comes on
three DVDs together.<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/genomewide_snp6_data.affx">http://www.affymetrix.com/support/technical/sample_data/genomewide_snp6_data.affx</a></p>

<p><strong>Affymetrix: Human Exon 1.0 ST Array Data Set</strong><br>
Chip type(s): HuEx-1_0-st-v2<br>
Description: Tissue/Mixture.<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/hugene_1_0_array_data.affx">http://www.affymetrix.com/support/technical/sample_data/hugene_1_0_array_data.affx</a></p>

<p><strong>Affymetrix: Human Gene 1.0 ST Array Data Set</strong><br>
Chip type(s): HuGene-1_0-st-v1<br>
Description: Tissues.<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/hugene_1_0_array_data.affx">http://www.affymetrix.com/support/technical/sample_data/hugene_1_0_array_data.affx</a></p>

<p><strong>Affymetrix: Mouse Gene 1.0 ST Array Data Set</strong><br>
Chip type(s):<br>
Description: Tissues.<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/hugene_1_0_array_data.affx">http://www.affymetrix.com/support/technical/sample_data/hugene_1_0_array_data.affx</a></p>

<p><strong>Affymetrix: Rat Gene 1.0 ST Array Data Set</strong><br>
Chip type(s):<br>
Description: Tissues.<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/hugene_1_0_array_data.affx">http://www.affymetrix.com/support/technical/sample_data/hugene_1_0_array_data.affx</a></p>

<p><strong>Affymetrix: Human Genome U133 Plus 2.0 Array Data Set</strong><br>
Chip type(s): HG-U133_Plus_2<br>
Description: Tissue/Mixture.<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/hugene_1_0_array_data.affx">http://www.affymetrix.com/support/technical/sample_data/hugene_1_0_array_data.affx</a></p>

<p><strong>Affymetrix: Mapping10K_Xba131/Mapping10K_Xba142</strong><br>
Chip type(s): Mapping10K_Xba131, Mapping10K_Xba142<br>
Description: CHP files (only) for 5 CEU trios (=15 individuals), 4
Africans, 2 Asians, 8 Caucasians.<br>
URL: <a href="https://www.affymetrix.com/analysis/10ksample.affx">https://www.affymetrix.com/analysis/10ksample.affx</a></p>

<p><strong>Affymetrix: Mapping 100k HapMap Trio Dataset</strong><br>
Chip type(s): Mapping50K_Xba240, Mapping50K_Hind240<br>
Description: Affymetrix - 30 CEU trios (= 30x3x2=180 hybridizations)<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/hapmap_trio_data.affx">http://www.affymetrix.com/support/technical/sample_data/hapmap_trio_data.affx</a></p>

<p><strong>Affymetrix: 14 Human Mitochondrial Resequencing samples</strong><br>
Chip type(s): Mitochip_2<br>
Description: Affymetrix.  This dataset contains 14 Mitov2 array CEL
files. The probe array files represent 3 sets of CEPH trios including 2
sets of technical replicates and 2 human mito standard samples.<br>
URL: <a href="http://www.affymetrix.com/support/datasets.affx">http://www.affymetrix.com/support/datasets.affx</a></p>

<p><strong>Affymetrix: 270 Mapping 100K HapMap samples</strong><br>
Chip type(s): Mapping50K_Xba240, Mapping50K_Hind240<br>
Description: HapMap. All 270 HapMap (CEU, CHB+JPT, YRI) 100K samples.<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/500k_data.affx">http://www.affymetrix.com/support/technical/sample_data/500k_data.affx</a></p>

<p><strong>Affymetrix: X Chromosome titration set for 100K</strong><br>
Chip type(s): Mapping50K_Xba240, Mapping50K_Hind240<br>
Description: Affymetrix - 1X, 2X, 3X, 4X, 5X titration set.<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/copy_number_data.affx#1_2">http://www.affymetrix.com/support/technical/sample_data/copy_number_data.affx#1_2</a></p>

<p><strong>Affymetrix:</strong> <strong>100K breast cancer cell lines</strong><br>
 Chip type(s): Mapping50K_Xba240, Mapping50K_Hind240<br>
Description: Affymetrix - 3 human breast cancer cell lines (SKBR3,
MCF7, &amp; ZR75-30), and reference DNA (Ref103).<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/copy_number_data.affx#1_2">http://www.affymetrix.com/support/technical/sample_data/copy_number_data.affx#1_2</a></p>

<p><strong>Affymetrix: 500K Breast cancer cell lines</strong><br>
Chip type(s): Mapping250K_Nsp, Mapping250K_Sty<br>
Description: Affymetrix - 3 human breast cancer cell lines (SKBR3,
MCF7, &amp; ZR75-30), and reference DNA (Ref103).<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/copy_number_data.affx">http://www.affymetrix.com/support/technical/sample_data/copy_number_data.affx</a></p>

<p><strong>Affymetrix: 48 Mapping 500k HapMap Trio Dataset</strong><br>
Chip type(s): Mapping250K_Nsp, Mapping250K_Sty<br>
Description: Affymetrix, 2005. 48 CEU trios (=13x3x2 Nsp+Sty 500K
hybridizations).<br>
URL: <a href="http://www.affymetrix.com/support/technical/sample_data/500k_data.affx">http://www.affymetrix.com/support/technical/sample_data/500k_data.affx</a></p>

<p><strong>Affymetrix: 270 Mapping 500k HapMap samples</strong><br>
Chip type(s): Mapping250K_Nsp, Mapping250K_Sty<br>
Description: HapMap, 2006. All 270 HapMap (CEU, CHB+JPT, YRI) 500K
samples.<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/500k_data.affx">http://www.affymetrix.com/support/technical/sample_data/500k_data.affx</a></p>

<p><strong>Affymetrix: Sample Data Sets for Copy Number Analysis</strong><br>
Chip type(s): Mapping250K_Nsp, Mapping250K_Sty<br>
Description: 9 Tumor/Normal pairs derived from human cancer cell lines
(=9x2 samples = 18x2 hybridizations), and X Chromosome titration set
(3X, 4X, and 5X) with 4 replicates. (=3x4x2 hybridizations). 2006.<br>
Note: The data sets are split up in multi-part ZIP archives.  See Page
&#39;<a href="http://groups.google.com/group/aroma-affymetrix/web/affymetrix-multi-part-dtt-zip-archives?hl=en">Affymetrix multi-part DTT/ZIP
archives</a>&#39;
on how to deal with such files.<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/copy_number_data.affx">http://www.affymetrix.com/support/technical/sample_data/copy_number_data.affx</a></p>

<p><strong>Affymetrix: Cytogenetics Sample Data</strong><br>
Chip type(s): GenomeWideSNP_6<br>
Description: 10 GWS6 CEL files of samples with UPD on chromosome 15
(two), DMD-del Xp21.1, Williams Syndrome, Mosaic Trisomy, Turner Mosaic,
Trisomy 13, Smith-Magenis, Angelman/Prader-Willi, and Normal Male
Sample.<br>
URL:
<a href="http://www.affymetrix.com/support/technical/sample_data/gtc_cytogenetic_data.affx">http://www.affymetrix.com/support/technical/sample_data/gtc_cytogenetic_data.affx</a></p>

<h2>Data sets from the HapMap Consortium</h2>

<p><strong>HapMap: HapMap Portal</strong><br>
URL: <a href="http://www.hapmap.org/"><strong>http://www.hapmap.org/</strong></a></p>

<p><strong>HapMap: 270</strong> <strong>Affymetrix</strong> <strong>Mapping 100K HapMap samples</strong><br>
Description: All 270 HapMap Phase II (CEU, CHB+JPT, YRI) 100K samples.<br>
URL: <a href="ftp://ftp.ncbi.nlm.nih.gov/hapmap/raw_data/affy100k/">ftp://ftp.ncbi.nlm.nih.gov/hapmap/raw_data/affy100k/</a></p>

<p><strong>HapMap: 270</strong> <strong>Affymetrix</strong> <strong>Mapping 500K HapMap samples</strong><br>
Chip type(s): Mapping250K_Nsp, Mapping250K_Sty<br>
Description: All 270 HapMap Phase II (CEU, CHB+JPT, YRI) 500K samples.<br>
URL: <a href="ftp://ftp.ncbi.nlm.nih.gov/hapmap/raw_data/affy500k/">ftp://ftp.ncbi.nlm.nih.gov/hapmap/raw_data/affy500k/</a></p>

<p><strong>HapMap: 1200+ Affymetrix GenomeWideSNP_6 HapMap samples</strong><br>
Chip type(s): GenomeWideSNP_6<br>
Description: 1200+ HapMap Phase III (many different populations) GWS6 samples.<br>
URL: <a href="ftp://ftp.ncbi.nlm.nih.gov/hapmap/raw_data/hapmap3_affy6.0">ftp://ftp.ncbi.nlm.nih.gov/hapmap/raw_data/hapmap3_affy6.0</a></p>

<h2>Data sets from EBI ArrayExpress</h2>

<p><strong>ArrayExpress: ArrayExpress Portal</strong><br>
URL:
<a href="http://www.ebi.ac.uk/arrayexpress/"><strong>http://www.ebi.ac.uk/arrayexpress/</strong></a></p>

<p><strong>ArrayExpress: Data set E-TABM-185</strong><br>
Chip type: HG-U133A<br>
Description: ArrayExpress, EMBL-EBI, March 2007.  5896 (sic!) HG-U133A
CEL files.<br>
URL: <a href="http://www.ebi.ac.uk/arrayexpress/#ae-browse/q=E-TABM-185">http://www.ebi.ac.uk/arrayexpress/#ae-browse/q=E-TABM-185</a></p>

<p><strong>ArrayExpress: Data set E-MEXP-1481</strong><br>
Chip type: Hs_PromPR_v02<br>
Title: Methylation profiling of human normal and cancer tissues to
identify long range epigenetic aberrations.<br>
Description: &quot;...25 hybridizations, using 25 samples of species [Homo
sapiens], using 25 arrays of Affymetrix GeneChip Human Promoter 1.0R
Array [Hs_PromPR_v02], producing 25 raw data files...&quot;<br>
URL:
<a href="http://www.ebi.ac.uk/microarray-as/ae/browse.html?keywords=E-MEXP-1481">http://www.ebi.ac.uk/microarray-as/ae/browse.html?keywords=E-MEXP-1481</a></p>

<h2>Data sets from the Public Expression Profiling Resource (PEPR)</h2>

<p><strong>PEPR: PEPR Portal</strong><br>
URL: <a href="http://pepr.cnmcresearch.org/">http://pepr.cnmcresearch.org/</a></p>

<h2>Data sets from NCBI&#39;s Gene Expression Omnibus (GEO)</h2>

<p><strong>GEO: Data Set Portal</strong></p>

<ul>
<li>  Search: <strong>Affymetrix data sets</strong><br>
URL:
<a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=gds&amp;term=affymetrix+AND+GPL%5BETYP%5D">http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=gds&amp;term=affymetrix+AND+GPL%5BETYP%5D</a></li>
<li>  Search: <strong>GenomeWideSNP_6 data sets</strong><br>
URL:
<a href="http://www.ncbi.nlm.nih.gov/sites/entrez?db=gds&amp;term=GPL6801%5BAccession%5D%20or%20GPL8226%5BAccession%5D&amp;cmd=search">http://www.ncbi.nlm.nih.gov/sites/entrez?db=gds&amp;term=GPL6801[Accession]
or
GPL8226[Accession]&amp;cmd=search</a></li>
</ul>

<p><strong>GSE5258: Connectivity Map dataset (build01)</strong><br>
Chip type(s): HG-U133A<br>
Description: &quot;A reference collection of genome-wide transcriptional
expression data for bioactive small molecules. [...] The current
collection (build01) contains data for 164 distinct small molecules
applied to freely cycling human cell lines, represented by 453
individual treatment and matched vehicle control pairs.&quot;.  In total 564
samples.<br>
Reference(s): Lamb J et al. The Connectivity Map: using gene-expression
signatures to connect small molecules, genes, and disease. Science,
2006.<br>
URL: <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE5258">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE5258</a></p>

<p><strong>GSE9222: Structural Variation of Chromosomes in Autism Spectrum
Disorder.</strong><br>
Chip type(s): Mapping250K<em>Nsp, Mapping250K_Sty<br>
Description: Mapping250K</em>{Nsp|Sty} CEL files for DNA derived from blood
or lymphoblasts. In total, 426 unrelated probands were analyzed along
with 232 parents (116 trios). ...<br>
Reference(s): Marshall et al. <em>Structural Variation of Chromosomes in
Autism Spectrum Disorder</em>, Am J Hum Genet, Feb 2008.<br>
URL: <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE9222">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE9222</a></p>

<p><strong>GSE12019: Fine-scale mapping of copy-number alterations with massively
parallel sequencing</strong><br>
Chip type(s): Mapping250K_Sty<br>
Description: &quot;In order to benchmark the reproducibility of Affymetrix
Genome-Wide Human SNP Array 6.0 for detecting copy-number alterations,
we performed replicate hybridizations of 3 tumor cell lines and 2 paired
normal cell lines obtained from the American Type Culture Collection
(ATCC). [...] 77 replicates of HCC1143 (breast ductal carcinoma), 69
replicates of HCC1143BL (matched normal), 42 replicates of HCC1954
(breast ductal carcinoma), 36 replicates of HCC1954BL (matched normal),
1 replicate of NCI-H2347 (lung adenocarcinoma)&quot;<br>
Reference(s): Chiang, D. Y.; Getz, G.; Jaffe, D. B.; O&#39;Kelly, M. J. T.;
Zhao, X.; Carter, S. L.; Russ, C.; Nusbaum, C.; Meyerson, M. &amp; Lander,
E. S. High-resolution mapping of copy-number alterations with massively
parallel sequencing. Nature Methods, 2009, 6, 99-103<br>
URL: <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE12019">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE12019</a></p>

<p><strong>GSE13021: Copy number analysis of human glioblastoma multiforme</strong><br>
Chip type(s): Mapping250K_Nsp<br>
Description: &quot;Copy number analysis of human GBM samples were performed,
and a high frequency of deletions of the PTPRD gene on chromosome
9p23-24.1 were identified. [...] Genomic DNA from 58 GBM tumor samples
were hybridized to Affymetrix 250K NspI Gene Chip Arrays and analyzed by
dChip using the hg17 genome assembly.&quot;<br>
Reference(s): Solomon DA, Kim JS, Cronin JC, Sibenaller Z et al.
Mutational inactivation of PTPRD in glioblastoma multiforme and
malignant melanoma. Cancer Res 2008 Dec 15;68(24):10300-6.<br>
URL: <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE13021">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE13021</a></p>

<p><strong>GSE13372: High-resolution mapping of copy-number alterations with
massively parallel sequencing</strong><br>
Chip type(s): GenomeWideSNP_6<br>
Description: &quot;In order to benchmark the reproducibility of Affymetrix
238K Sty arrays for detecting copy-number alterations. We performed
replicate hybridizations of 3 tumor cell lines and 2 paired normal cell
lines obtained from the American Type Culture Collection (ATCC). [...]
21 replicates of HCC1143 (breast ductal carcinoma), 21 replicates of
HCC1143BL (matched normal), 13 replicates of HCC1954 (breast ductal
carcinoma), 11 replicates of HCC1954BL (matched normal), 1 replicate of
NCI-H2347 (lung adenocarcinoma), 1 replicate of NCI-H2347BL (matched
normal)&quot;<br>
Reference(s): Chiang, D. Y.; Getz, G.; Jaffe, D. B.; O&#39;Kelly, M. J. T.;
Zhao, X.; Carter, S. L.; Russ, C.; Nusbaum, C.; Meyerson, M. &amp; Lander,
E. S. High-resolution mapping of copy-number alterations with massively
parallel sequencing. Nature Methods, 2009, 6, 99-103<br>
URL: <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE13372">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE13372</a><br>
URL: <a href="http://www.broad.mit.edu/cancer/pub/solexa_copy_numbers/">http://www.broad.mit.edu/cancer/pub/solexa_copy_numbers/</a></p>

<p><strong>GSE14996: Multisampled Lethal Metastatic Prostate Cancer Copy Number
Analysis</strong><br>
Chip type(s): GenomeWideSNP_6<br>
Description: &quot;Purified cancer DNA from frozen metastatic cancer tissue
obtained at autopsy studied by Affymetrix Genome-Wide Human SNP (single
nucleotide polymorphism) Array 6.0 analysis (Affy6).&quot;<br>
Reference(s): Liu W, Laitinen S, Khan S, Vihinen M et al. Copy number
analysis indicates monoclonal origin of lethal metastatic prostate
cancer. Nat Med 2009 May;15(5):559-65.<br>
URL: <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE14996">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE14996</a></p>

<p><strong>GSE16619: Identification of novel gene amplification events in breast
cancer</strong><br>
Chip type(s): Mapping250K_Nsp, Mapping250K_Sty, GenomeWideSNP_5<br>
Description: To identify novel gene amplification events that may
contribute to breast cancer progression, we examined copy number
variation in 161 primary breast cancer samples using the Affymetrix
250K_Nsp and 250K_Sty microarrays or the Affymetrix SNP5.0
microarray.<br>
Reference(s): Kadota M, Sato M, Duncan B, Ooshima A et al.
Identification of novel gene amplifications in breast cancer and
coexistence of gene amplification with an activating mutation of PIK3CA.
Cancer Res 2009 Sep 15;69(18):7357-65. PMID: 19706770.<br>
URL: <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE16619">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE16619</a></p>

<p><strong>GSE20584: The mutation spectrum revealed by paired genome sequences
from a lung cancer patient</strong><br>
Chip type(s): GenomeWideSNP_6<br>
Description: One lung tumor and its adjacent normal were profiled for copy-number alterations with the high-resolution Affymetrix SNP6.0 Array.<br>
Reference(s): Lee W, Jiang Z, Liu J, Haverty PM et al. The mutation spectrum revealed by paired genome sequences from a lung cancer patient. Nature 2010 May 27;465(7297):473-7. PMID: <a href="http://www.ncbi.nlm.nih.gov/sites/entrez?Db=Pubmed&amp;term=20505728%5BUID%5D" title="Link to PubMed record">20505728</a><br>
URL: <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE20584">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE20584</a></p>

<h2>Data sets from the Tumor Cancer Genome Atlas (TCGA) project</h2>

<p>TCGA: TCGA Data Portal<br>
Description: The Cancer Genome Atlas (TCGA) project.<br>
URL: <a href="http://tcga-data.nci.nih.gov/tcga/homepage.htm">http://tcga-data.nci.nih.gov/tcga/homepage.htm</a> (Data Access
Matrix)<br>
URL: <a href="http://tcga-data.nci.nih.gov/tcga/findArchives.htm">http://tcga-data.nci.nih.gov/tcga/findArchives.htm</a> (Search by
Archives; old approach)</p>

<h2>Data sets from The Cancer Genome Project (Sanger Institute)</h2>

<p>URL: <a href="http://www.sanger.ac.uk/genetics/CGP/">http://www.sanger.ac.uk/genetics/CGP/</a>
CGP: CGP Data Archive<br>
Chip type(s): GenomeWideSNP_6<br>
Description: The Cancer Genome Project data repository.<br>
URL: <a href="http://www.sanger.ac.uk/genetics/CGP/Archive/">http://www.sanger.ac.uk/genetics/CGP/Archive/</a></p>

<h2>Data sets from The Broad Institute</h2>

<p>Broad Institute et al.: Broad-Novartis Cancer Cell Line Encyclopedia
(CCLE)<br>
 Chip type(s): GenomeWideSNP_6, HG-U133_Plus_2, OncoMap mutation
data, Hybrib capture sequencing<br>
 Description: &quot;The Cancer Cell Line Encyclopedia (CCLE) project is a
collaboration between the Broad Institute, and the Novartis Institutes
for Biomedical Research and its Genomics Institute of the Novartis
Research Foundation to conduct a detailed genetic and pharmacologic
characterization of a large panel of human cancer models, to develop
integrated computational analyses that link distinct pharmacologic
vulnerabilities to genomic patterns and to translate cell line
integrative genomics into cancer patient stratification. The CCLE
provides public access to genomic data, analysis and visualization for
about 1000 cell lines.&quot;<br>
 URL: <a href="http://www.broadinstitute.org/ccle/">http://www.broadinstitute.org/ccle/</a></p>

<p>Broad Institute: GenomeWideSNP_6 test samples for Birdsuite<br>
Chip type(s): GenomeWideSNP_6<br>
URL: <a href="http://www.broad.mit.edu/mpg/birdsuite/download.html">http://www.broad.mit.edu/mpg/birdsuite/download.html</a><br>
Description: 3 male and 3 female GenomeWideSNP_6 CEL files (part of the
&quot;Test input files&quot;).</p>

<p>Broad Institute: Genotyping the 270 HapMap samples for GAIN<br>
URL:
<a href="http://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?id=phs000049">http://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?id=phs000049</a><br>
Description: 270 HapMap samples data set [270 hybridizations] - No more
details.</p>

<p>Broad Institute: Assessing the significance of chromosomal aberrations
in cancer<br>
 Chip type(s): Mapping50K_Hind240, Mapping50K_Xba240<br>
 Description: Brain Cancer, SNP/CN Analysis.  154 glioma and 33 normal
samples each hybridized on the 100K chip set totalling (154+33)*2=374
hybridizations/CEL files.  Also available at GEO as
<a href="http://www.ncbi.nlm.nih.gov/projects/geo/query/acc.cgi?acc=GSE9635">GSE9635</a>.<br>
 Reference: Beroukhim et al. 2007, Assessing the significance of
chromosomal aberrations in cancer: Methodology and application to
glioma, PNAS, December 2007.<br>
 URL:
<a href="http://www.broad.mit.edu/cgi-bin/cancer/publications/pub_paper.cgi?mode=view&amp;paper_id=162&amp;p=t">http://www.broad.mit.edu/cgi-bin/cancer/publications/pub_paper.cgi?mode=view&amp;paper_id=162&amp;p=t</a></p>

<p>Broad Institute et al.: Characterizing the cancer genome in lung
adenocarcinoma<br>
 Chip type(s): Mapping250K_Sty<br>
 Description: 2*384=768 Affymetrix Mapping250K_Sty CEL files for 384
tumor/normal sample pairs for Weir et al. (2007).<br>
 Reference: Weir et al. Characterizing the cancer genome in lung
adenocarcinoma. Nature, December 2007, 450, 893-898.<br>
 URL: <a href="http://www.broad.mit.edu/cancer/pub/tsp/">http://www.broad.mit.edu/cancer/pub/tsp/</a></p>

<p>Broad Institute et al.: Connectivity Map (02)<br>
 Chip type(s): HG-U133A<br>
 Description: 6100 Affymetrix HG-U133A CEL files.<br>
 Reference: (1) Lamb et al., The Connectivity Map: using gene-expression
signatures to connect small molecules, genes, and disease, Science,
2006. (2) J. Lamb, The Connectivity Map: a new tool for biomedical
research, Nat Rev Cancer, 2007.<br>
 URL: <a href="http://www.broadinstitute.org/cmap/">http://www.broadinstitute.org/cmap/</a></p>

<h2>Data sets from NIH&#39;s REMBRANDT</h2>

<p>&quot;REpository for Molecular BRAin Neoplasia DaTa (REMBRANDT) is a robust
bioinformatics knowledgebase framework that leverages data warehousing
technology to host and integrate clinical and functional genomics data
from clinical trials involving patients suffering from Gliomas. The
knowledge framework will provide researchers with the ability to perform
ad hoc querying and reporting across multiple data domains, such as Gene
Expression, Chromosomal aberrations and Clinical data.&quot;<br>
URL: <a href="https://caintegrator.nci.nih.gov/rembrandt/">https://caintegrator.nci.nih.gov/rembrandt/</a></p>

<h2>Data sets from elsewhere</h2>

<p>GSK Cancer Cell Line Genomic Profiling Data<br>
Chip types: Mapping 250K_Nsp, Mapping 250K_Sty, HG-U133_Plus_2<br>
Description: &quot;GlaxoSmithKline (GSK) has released the genomic profiling
data for over 300 cancer cell lines via the National Cancer Institute&#39;s
cancer Bioinformatics Grid (caBIG). Cancer cell lines can be
manipulated in the laboratory and have been used extensively by GSK in
the discovery and development of novel cancer therapeutics.&quot;<br>
Data files: CEL files for in total 676 Affymetrix Mapping 250K_Nsp or
Mapping 250K_Sty, and 950 HG-U133_Plus_2 hybridizations.  Also MAS5
summarizes for the expression arrays.<br>
URL: <a href="https://cabig.nci.nih.gov/tools/caArray_GSKdata/">https://cabig.nci.nih.gov/tools/caArray_GSKdata/</a></p>

<p>Harvard: The Meyerson Lab<br>
Chip types: ax13339 (&quot;Early Access 10K&quot;)<br>
Description: 58 paired CEL and *.txt genotype files for the ax13339
(&quot;Early Access 10K&quot;) used in Zhao X et al. (2004).<br>
References: Zhao X et al., An Integrated View of Copy Number and Allelic
Alterations in the Cancer Genome Using Single Nucleotide Polymorphism
Arrays, Cancer Research, 64, 3060-3071, May 2004.<br>
URL: <a href="http://research.dfci.harvard.edu/meyersonlab/snp/snp.htm">http://research.dfci.harvard.edu/meyersonlab/snp/snp.htm</a> (under
&#39;Data download&#39;)</p>

<p>Personal Genome Project (PGP)<br>
Description: Currently there are 10 individuals for which PGP are doing
SNP, CN, and exon analysis on.  It is not clear if it is ony the results
or the raw data that is/will be available.  It looks like they will
publish many more &quot;personal genomes&quot; later.<br>
URL: <a href="http://www.personalgenomes.org/public/">http://www.personalgenomes.org/public/</a></p>

<h2>Miscellaneous</h2>

<p>ACTuDB: a database for the integrated analysis of array-CGH and clinical
data for tumors<br>
Author(s): The Curie Institute - Bioinformatics Unit<br>
URL: <a href="http://bioinfo-out.curie.fr/actudb/">http://bioinfo-out.curie.fr/actudb/</a> (see the &#39;Content&#39; section)<br>
Description: Contains a compiled list of publicly available
copy-number data sets.  Some of the data sets also have coupled
expression data.
<a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL6801">GPL6801</a> or <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL8226">GPL8226</a></p>
</div>
</div> 
</div> 

<div id="filler"></div>


<div id="push"></div> 
<div id="footer" style="height: auto;">
  <div class="container">
<p class="muted credit" style="font-size: small;">
  The Aroma Project
  <span class="printonly">(http://www.aroma-project.org/)</span>
  ::
  &copy; Henrik Bengtsson
  ::
  Powered by <a href="http://cran.r-project.org/package=R.rsp">RSP</a>

  <span style="float: right;">
  <a href="http://validator.w3.org/check?st=1;outline=1;verbose=1;group=1&amp;uri=http://www.aroma-project.org/datasets/Affymetrix/" aria-label="Validate HTML" id="validate-page" target="_blank"><span class="glyphicon glyphicon-check" title="Validate HTML" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Validate HTML</span></a>
  &nbsp;
  <a href="https://github.com/AromaProject/aroma.project.org-website/tree/master/content/datasets/Affymetrix/index.md.rsp" id="edit"
  target="_blank"><span class="glyphicon glyphicon-edit" title="Edit page" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Edit page</span></a>
  &nbsp;
  <a href="http://v1.aroma-project.org/datasets/Affymetrix/" id="old-page"
  target="_blank"><span class="glyphicon glyphicon-time" title="Go to old page" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Go to old page</span></a>
  </span>
</p>

  </div>
</div>

<div class="device-xs visible-xs"></div>
<div class="device-sm visible-sm"></div>
<div class="device-md visible-md"></div>
<div class="device-lg visible-lg"></div>

<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
<script src="../../assets/js/post-jquery.js"></script>
<script src="http://netdna.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
<script src="../../assets/js/post-bootstrap.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  
    ga('create', 'UA-11999401-1', 'auto');
    ga('send', 'pageview');
  </script>
</body>
</html>
