<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="keywords" content="R, Bioconductor, Affymetrix">
  <meta name="author" content="Henrik Bengtsson">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="shortcut icon" type="image/png" href="../../assets/ico/favicon.png">
  <title title="The Aroma Project (Aroma): TumorBoost - Normalization of allelic-specific copy numbers in tumors with matched normals">Aroma: TumorBoost - Normalization of allelic-specific copy numbers in tumors with matched normals</title>
  <link href="../../assets/css/bootstrap.min.css" rel="stylesheet">
  <link href="../../assets/css/cbc-tweaks.css" rel="stylesheet">
  <link href="../../assets/css/r_highlight.css" rel="stylesheet">
  <script src="../../assets/js/r_highlight.js"></script>
</head>
<body>
<a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>
<nav class="navbar navbar-default navbar-fixed-top" role="navigation" id="topnavbar">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../../index.html" title="The Aroma Project">Aroma</a>
    </div>

    <div class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../../getstarted/index.html">Get Started</a></li>
        <li><a href="../../docs/index.html">Documentation</a></li>
        <li><a href="../../features/index.html">Features</a></li>
        <li><a href="../../resources/index.html">Resources</a></li>
        <li><a href="../../forum/index.html">Forum</a></li>
        <li><a href="../../FAQ/index.html">FAQ</a></li>
        <li><a href="../../developers/index.html">Developers' Corner</a></li>
        <li><a href="../../about/index.html">About</a></li>
      </ul>
    </div>
  </div>
</nav>

<div id="wrap">
<div class="bs-docs-header" id="content">
<div class="container">
<h1>TumorBoost - Normalization of allelic-specific copy numbers in tumors with matched normals</h1>

<p>Author: Pierre Neuvial<br>
Created on: 2010-05-30<br>
Last updated: 2011-03-08</p>

<p><img src="../../assets/images/TumorBoost/BeadStudio%2CXY%2CTCGA-02-0001%2Ctumor%2CChr02%2CfracB.png" alt="">  </p>

<p><img src="../../assets/images/TumorBoost/BeadStudio%2CXY%2CTCGA-02-0001%2CtumorN%2CChr02%2CfracB.png" alt=""></p>

<p><em>Figure: Allele B fractions before and after TumorBoost normalization.</em></p>

<h2>Introduction</h2>

<p>This vignette illustrates the TumorBoost method for normalizing allelic
ratios from genotyping microarrays, which is described in
<a name=cite-BengtssonH_etal_2010-TumorBoost></a><a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;amp;cmd=prlinks&amp;amp;retmode=ref&amp;amp;id=20462408">Bengtsson, Neuvial, and Speed (2010)</a>.
The input of the method is a single tumor-normal pair of
allelic ratios from genotyping microarrays.</p>

<pre><code>totalAndFracBData/
  TCGA,GBM,BeadStudio,XY/
    HumanHap550/
      TCGA-02-0001-01C-01D-0184-06,fracB.asb
      TCGA-02-0001-01C-01D-0184-06,total.asb
      TCGA-02-0001-10A-01D-0184-06,fracB.asb
      TCGA-02-0001-10A-01D-0184-06,total.asb

  TCGA,GBM,CRMAv2/
    GenomeWideSNP/
      TCGA-02-0001-01C-01D-0182-01,fracB.asb
      TCGA-02-0001-01C-01D-0182-01,total.asb
      TCGA-02-0001-10A-01D-0182-01,fracB.asb
      TCGA-02-0001-10A-01D-0182-01,total.asb

callData/
  TCGA,GBM,BeadStudio,XY,NGC/
    HumanHap550/
      TCGA-02-0001,10A,01D-0184-06,genotypes,confidenceScores.acf
      TCGA-02-0001,10A,01D-0184-06,genotypes.acf

annotationData/
  chipTypes/
    HumanHap550/
      HumanHap550,TCGA,HB20080512.ugp
      HumanHap550,TCGA,HB20100107,unitNames.txt

    GenomeWideSNP_6/
      GenomeWideSNP_6,Full.CDF
      GenomeWideSNP_6,Full,na26,HB20080821.ugp
</code></pre>

<p>Note: In the above data sets, files corresponding to both allelic ratios
(&#39;fracB&#39;) and total intensities (&#39;total&#39;) are listed.  Total intensities
are only used at the end of this vignette to display total copy number
profiles. Total intensities are not used for TumorBoost normalization,
and total copy numbers are not affected by TumorBoost normalization.</p>

<h2>Input data: genotype calls</h2>

<p>To normalize a tumor sample, TumorBoost uses genotype calls (and
optionally genotype confidence scores) of the matched normal sample.  In
this vignette we assume that such genotype calls are available to us
(see &#39;callData&#39; above). In the vignette &#39;<a href="../../vignettes/naive-genotyping/index.html">Naive genotype calls and
associated confidence scores</a>&#39; we show how
such genotype calls can be estimated only based on allelic ratios in the
normal sample, as described in the TumorBoost paper.</p>

<p>To use other genotype calls than the naive ones, a data set with similar
structure as above has to be created, for example:</p>

<pre><code>callData/
  TCGA,GBM,BeadStudio,XY,BeadStudio/
    HumanHap550/
      TCGA-02-0001,10A,01D-0184-06,genotypes,confidenceScores.acf
      TCGA-02-0001,10A,01D-0184-06,genotypes.acf
</code></pre>

<p>The creation of such a data set depends on the input data (Birdseed or
BeadStudio genotypes for example).  It is not documented here.</p>

<h2>Input data: allelic ratios</h2>

<p>For Affymetrix genotyping microarrays, input data can be obtained using
the CRMA v2 method <a name=cite-BengtssonH_etal_2009-CRMAv2></a>(<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;amp;cmd=prlinks&amp;amp;retmode=ref&amp;amp;id=19535535">Bengtsson, Wirapati, and Speed, 2009</a>), e.g.</p>

<pre><code class="r">ds &lt;- doASCRMAv2(&quot;TCGA,GBM&quot;, chipType=&quot;GenomeWideSNP_6,Full&quot;)
</code></pre>

<p>See <a href="../../blocks/doCRMAv2/index.html">CRMA v2 block</a> and <a href="../../vignettes/CRMAv2/index.html">CRMA v2
vignette</a> for details.</p>

<h2>Setup</h2>

<pre><code class="r">library(&quot;aroma.cn&quot;)
library(&quot;R.devices&quot;)
devOptions(&quot;png&quot;, width=1024)

log &lt;- verbose &lt;- Arguments$getVerbose(-8, timestamp=TRUE)
rootPath &lt;- &quot;totalAndFracBData&quot;
rootPath &lt;- Arguments$getReadablePath(rootPath)
dataSets &lt;- c(&quot;TCGA,GBM,BeadStudio,XY&quot;, &quot;TCGA,GBM,CRMAv2&quot;)

dataSet &lt;- dataSets[1] ## Work with Illumina data

# Fullnames translator
fnt &lt;- function(names, ...) {
  pattern &lt;- &quot;^(TCGA-[0-9]{2}-[0-9]{4})-([0-9]{2}[A-Z])[-]*(.*)&quot;
  gsub(pattern, &quot;\\1,\\2,\\3&quot;, names)
} # fnt()
</code></pre>

<h2>Load the raw (tumor,normal) data set</h2>

<p>Here we will assume that matching tumors and normals share the same name
but have different tags, e.g. TCGA-02-0001,01C,01D-0184-06 (tumor)
and TCGA-02-0001,10A,01D-0184-06 (normal) with the common
name TCGA-02-0001.</p>

<pre><code class="r">ds &lt;- AromaUnitFracBCnBinarySet$byName(dataSet, chipType=&quot;*&quot;, paths=rootPath)
setFullNamesTranslator(ds, fnt)
print(ds)

# Identify all sample names by extracting all unique arrays name (ignoring tags)
sampleNames &lt;- sort(unique(getNames(ds))) # Work with the first sample only
sampleName &lt;- sampleNames[1]
# Extract the two arrays with this name, which should be the tumor and the normal
pair &lt;- indexOf(ds, sampleName)
stopifnot(length(pair) == 2)

# Order as (tumor,normal)
types &lt;- sapply(ds[pair], FUN=function(df) getTags(df)[1])
o &lt;- order(types)
types &lt;- types[o]
pair &lt;- pair[o]

# Extract (tumor, normal) pair
dsPair &lt;- ds[pair]
dsT &lt;- dsPair[1]
print(dsT)

dsN &lt;- dsPair[2]
print(dsN)
</code></pre>

<p>This gives:</p>

<pre><code>AromaUnitFracBCnBinarySet:
Name: TCGA
Tags: GBM,BeadStudio,XY
Full name: TCGA,GBM,BeadStudio,XY
Number of files: 1
Names: TCGA-02-0001 [1]
Path (to the first file): totalAndFracBData/TCGA,GBM,BeadStudio,XY/HumanHap550
Total file size: 2.14 MB
RAM: 0.00MB
</code></pre>

<h2>Load the genotype call set</h2>

<pre><code class="r"># Identify available genotype calls
rootPath &lt;- &quot;callData&quot;
rootPath &lt;- Arguments$getReadablePath(rootPath)

genotypeTag &lt;- &quot;NGC&quot;
gsN &lt;- AromaUnitGenotypeCallSet$byName(dataSet, tags=genotypeTag, chipType=&quot;*&quot;)
setFullNamesTranslator(gsN, fnt)

# Keep only normal genotype files (not needed here, but could be needed in other situations)
types &lt;- sapply(gsN, FUN=function(df) getTags(df)[1])
types &lt;- gsub(&quot;[A-Z]$&quot;, &quot;&quot;, types)
keep &lt;- which(is.element(types, c(&quot;10&quot;, &quot;11&quot;)))
gsN &lt;- gsN[keep]
print(gsN)
</code></pre>

<p>This gives:</p>

<pre><code>AromaUnitGenotypeCallSet:
Name: TCGA
Tags: GBM,BeadStudio,XY,NGC
Full name: TCGA,GBM,BeadStudio,XY,NGC
Number of files: 1
Names: TCGA-02-0001 [1]
Path (to the first file): callData/TCGA,GBM,BeadStudio,XY,NGC/HumanHap550
Total file size: 1.07 MB
RAM: 0.00MB
</code></pre>

<h2>Apply TumorBoost normalization</h2>

<p>Recall that we in this vignette assume that matching tumors and normals
share the same name but have different tags,
e.g. TCGA-02-0001,01C,01D-0184-06 (tumor)
and TCGA-02-0001,10A,01D-0184-06 (normal) with the common
name TCGA-02-0001.  In what follows, we utilize this fact by sorting
three data sets such that they are ordered in the same way (as the
normal set).</p>

<p>If your tumors and normals do not share the same names this way, you
have to make sure the sets are ordered correctly after this step in
order for the rest of the analysis to be correct.</p>

<pre><code class="r"># Create a list of matched data sets
dsList &lt;- list(normal=dsN, tumor=dsT, callsN=gsN)
# Make sure they are ordered the same way such that the
# k:th array corresponds to the same sample in all sets.
sampleNames &lt;- getNames(dsList$normal)
dsList &lt;- lapply(dsList, FUN=function(ds) {
  idxs &lt;- indexOf(ds, sampleNames)
  ds[idxs]
})
print(dsList)
</code></pre>

<p>This gives:</p>

<pre><code>$normal
AromaUnitFracBCnBinarySet:
Name: TCGA
Tags: GBM,BeadStudio,XY
Full name: TCGA,GBM,BeadStudio,XY
Number of files: 1
Names: TCGA-02-0001 [1]
Path (to the first file): totalAndFracBData/TCGA,GBM,BeadStudio,XY/HumanHap550
Total file size: 2.14 MB
RAM: 0.00MB

$tumor
AromaUnitFracBCnBinarySet:
Name: TCGA
Tags: GBM,BeadStudio,XY
Full name: TCGA,GBM,BeadStudio,XY
Number of files: 1
Names: TCGA-02-0001 [1]
Path (to the first file): totalAndFracBData/TCGA,GBM,BeadStudio,XY/HumanHap550
Total file size: 2.14 MB
RAM: 0.00MB

$callsN
AromaUnitGenotypeCallSet:
Name: TCGA
Tags: GBM,BeadStudio,XY,NGC
Full name: TCGA,GBM,BeadStudio,XY,NGC
Number of files: 1
Names: TCGA-02-0001 [1]
Path (to the first file): callData/TCGA,GBM,BeadStudio,XY,NGC/HumanHap550
Total file size: 1.07 MB
RAM: 0.00MB
</code></pre>

<pre><code class="r">dummy &lt;- lapply(dsList, FUN=function(ds) print(ds[[1]]))

# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
# Normalize allele B fractions for tumors given matched normals
# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
tbn &lt;- TumorBoostNormalization(dsList$tumor, dsList$normal, gcN=dsList$callsN, tags=c(&quot;*&quot;, &quot;NGC&quot;))
dsTN &lt;- process(tbn, verbose=log)
setFullNamesTranslator(dsTN, fnt)
print(dsTN)
</code></pre>

<p>This gives:</p>

<pre><code>AromaUnitFracBCnBinarySet:
Name: TCGA
Tags: GBM,BeadStudio,XY,TBN,NGC
Full name: TCGA,GBM,BeadStudio,XY,TBN,NGC
Number of files: 1
Names: TCGA-02-0001[1]
Path (to the first file): totalAndFracBData/TCGA,GBM,BeadStudio,XY,TBN,NGC/HumanHap550
Total file size: 2.14 MB
RAM: 0.00MB
</code></pre>

<p>This has created the following data set:</p>

<pre><code>totalAndFracBData/
  TCGA,GBM,BeadStudio,XY,TBN,NGC/
    HumanHap550/
      TCGA-02-0001-01C-01D-0184-06,fracB.asb
</code></pre>

<h3>Create a list of matched data sets</h3>

<pre><code class="r">dsList &lt;- list(normal=dsN, tumor=dsT, tumorN=dsTN, callsN=gsN)
dsList &lt;- lapply(dsList, FUN=function(ds) {
  idxs &lt;- indexOf(ds, getNames(dsList$normal))
  ds[idxs]
})
</code></pre>

<h3>Plots</h3>

<pre><code class="r">figPath &lt;- Arguments$getWritablePath(&quot;figures&quot;)
siteTag &lt;- getTags(ds)
siteTag &lt;- paste(siteTag[-1], collapse=&quot;,&quot;)
print(siteTag)

## [1] &quot;BeadStudio,XY&quot;

ugp &lt;- getAromaUgpFile(dsList$tumor)

chromosome &lt;- 2
chrTag &lt;- sprintf(&quot;Chr%02d&quot;, chromosome)
units &lt;- getUnitsOnChromosome(ugp, chromosome=chromosome)

# Identify SNPs only
platform &lt;- getPlatform(ugp)
if (platform == &quot;Affymetrix&quot;) {
  require(&quot;aroma.affymetrix&quot;) || throw(&quot;Package not loaded: aroma.affymetrix&quot;)
  snpPattern &lt;- &quot;^SNP&quot;
} else if (platform == &quot;Illumina&quot;) {
  snpPattern &lt;- &quot;^rs[0-9]&quot;
} else {
  throw(&quot;Unknown platform: &quot;, platform)
}
unf &lt;- getUnitNamesFile(ugp)
unitNames &lt;- getUnitNames(unf, units=units)

# Identify SNP units
keep &lt;- (regexpr(snpPattern, unitNames) != -1)
units &lt;- units[keep]

pos &lt;- getPositions(ugp, units=units)

# Extract Allele B fractions
kk &lt;- 1
dfList &lt;- lapply(dsList, FUN=`[[`, kk)
beta &lt;- lapply(dfList, FUN=function(df) df[units,1,drop=TRUE])
beta &lt;- as.data.frame(beta)
beta &lt;- as.matrix(beta)
names &lt;- colnames(beta)
names[names == &quot;tumorN&quot;] &lt;- &quot;normalized tumor&quot;

# Plot dimensions
x &lt;- pos/1e6
xlim &lt;- range(x, na.rm=TRUE)
xlab &lt;- &quot;Position (Mb)&quot;
</code></pre>

<h3>Plot allele B fractions</h3>

<pre><code class="r">cols &lt;- as.integer(beta[,&quot;callsN&quot;] != 1) + 1L

for (cc in 1:3) {
  toPNG(sampleName, tags=c(siteTag, colnames(beta)[cc], chrTag, &quot;fracB&quot;), aspectRatio=0.6*1/3, {
    par(mar=c(1.7,2.5,1.1,1)+0.1, tcl=-0.3, mgp=c(1.4,0.4,0), cex=2)
    plot(NA, xlim=xlim, ylim=c(-0.1,1.1), xlab=xlab, ylab=&quot;Allele B Fraction&quot;, axes=FALSE)
    axis(side=1)
    axis(side=2, at=c(0,1/2,1))
    points(x, beta[,cc], pch=&quot;.&quot;, col=cols)
    label &lt;- sprintf(&quot;%s (%s)&quot;, sampleName, names[cc])
    stext(side=3, pos=0, label)
    stext(side=3, pos=1, chrTag)
  })
}
</code></pre>

<h3>Plot total CN</h3>

<pre><code class="r">rootPath &lt;- &quot;totalAndFracBData&quot;
rootPath &lt;- Arguments$getReadablePath(rootPath)
dsC &lt;- AromaUnitTotalCnBinarySet$byName(dataSet, chipType=&quot;*&quot;, paths=rootPath)
setFullNamesTranslator(dsC, fnt)
print(dsC)
</code></pre>

<p>This gives:</p>

<pre><code>AromaUnitTotalCnBinarySet:
Name: TCGA
Tags: GBM,BeadStudio,XY
Full name: TCGA,GBM,BeadStudio,XY
Number of files: 2
Names: TCGA-02-0001, TCGA-02-0001 [2]
Path (to the first file): totalAndFracBData/TCGA,GBM,BeadStudio,XY/HumanHap550
Total file size: 4.28 MB
RAM: 0.00MB
</code></pre>

<pre><code class="r">pairC &lt;- indexOf(dsC, sampleName)
stopifnot(length(pairC) == 2)

# Order as (tumor,normal)
types &lt;- sapply(dsC[pairC], FUN=function(df) getTags(df)[1])
o &lt;- order(types)
types &lt;- types[o]
pairC &lt;- pairC[o]

# Extract (tumor, normal) pair
dsPairC &lt;- dsC[pairC]

# Extract total CNs
C &lt;- extractMatrix(dsPairC, units=units)
C &lt;- 2*C[,1]/C[,2]

# Plot total CNs
toPNG(sampleName, tags=c(siteTag, chrTag, &quot;CN&quot;), aspectRatio=0.6*1/3, {
  par(mar=c(1.7,2.5,1.1,1)+0.1, tcl=-0.3, mgp=c(1.4,0.4,0), cex=2)
  plot(NA, xlim=xlim, ylim=c(0,6), xlab=xlab, ylab=&quot;Copy number&quot;, axes=FALSE)
  axis(side=1)
  axis(side=2, at=c(0,2,4,6))
  points(x, C, pch=&quot;.&quot;)
  label &lt;- sprintf(&quot;%s&quot;, sampleName)
  stext(side=3, pos=0, label)
  stext(side=3, pos=1, chrTag)
})
</code></pre>

<p>The images generated above are reproduced in Figure 1.</p>

<p><img src="../../assets/images/TumorBoost/BeadStudio%2CXY%2CTCGA-02-0001%2CChr02%2CCN.png" alt="">  </p>

<p><img src="../../assets/images/TumorBoost/BeadStudio%2CXY%2CTCGA-02-0001%2Cnormal%2CChr02%2CfracB.png" alt="">  </p>

<p><img src="../../assets/images/TumorBoost/BeadStudio%2CXY%2CTCGA-02-0001%2Ctumor%2CChr02%2CfracB.png" alt="">  </p>

<p><img src="../../assets/images/TumorBoost/BeadStudio%2CXY%2CTCGA-02-0001%2CtumorN%2CChr02%2CfracB.png" alt=""></p>

<p><em>Figure 1. Total copy numbers (top panel), Allele B fractions in the
normal (second panel), in the tumor (third panel), and in the normalized
tumor (bottom panel).  Data are from the Illumina HumanHap550 platform,
preprocessed with BeadStudio (method &quot;XY&quot;, not &quot;BAF&quot;).</em></p>

<h2>References</h2>

<p><a
name=bib-BengtssonH_etal_2009-CRMAv2></a><a href="#cite-BengtssonH_etal_2009-CRMAv2">[1]</a>
H. Bengtsson, P. Wirapati and T. P. Speed. &quot;A single-array
preprocessing method for estimating full-resolution raw copy numbers
from all Affymetrix genotyping arrays including GenomeWideSNP 5 &amp; 6&quot;.
Eng. In: <em>Bioinformatics (Oxford, England)</em> 25.17 (2009), pp. 2149-56.
DOI:
<a href="http://dx.doi.org/10.1093/bioinformatics/btp371">10.1093/bioinformatics/btp371</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=19535535">19535535</a>.</p>

<p><a
name=bib-BengtssonH_etal_2010-TumorBoost></a><a href="#cite-BengtssonH_etal_2010-TumorBoost">[2]</a>
H. Bengtsson, P. Neuvial and T. P. Speed. &quot;TumorBoost: normalization of
allele-specific tumor copy numbers from a single pair of tumor-normal
genotyping microarrays&quot;. Eng. In: <em>BMC bioinformatics</em> 11 (2010), p.
245. DOI:
<a href="http://dx.doi.org/10.1186/1471-2105-11-245">10.1186/1471-2105-11-245</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=20462408">20462408</a>.</p>
</div>
</div> 
</div> 

<div id="filler"></div>


<div id="push"></div> 
<div id="footer" style="height: auto;">
  <div class="container">
<p class="muted credit" style="font-size: small;">
  The Aroma Project
  <span class="printonly">(http://www.aroma-project.org/)</span>
  ::
  Â© Henrik Bengtsson
  ::
  Powered by <a href="http://cran.r-project.org/package=R.rsp">RSP</a>

  <span style="float: right;">
  <a href="http://validator.w3.org/check?st=1;outline=1;verbose=1;group=1&amp;uri=http://www.aroma-project.org/vignettes/tumorboost-highlevel/" aria-label="Validate HTML" id="validate-page" target="_blank"><span class="glyphicon glyphicon-check" title="Validate HTML" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Validate HTML</span></a>
  &nbsp;
  <a href="https://github.com/AromaProject/aroma.project.org-website/tree/master/content/vignettes/tumorboost-highlevel/index.md.rsp" id="edit"
  target="_blank"><span class="glyphicon glyphicon-edit" title="Edit page" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Edit page</span></a>
  &nbsp;
  <a href="http://v1.aroma-project.org/vignettes/tumorboost-highlevel/" id="old-page"
  target="_blank"><span class="glyphicon glyphicon-time" title="Go to old page" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Go to old page</span></a>
  </span>
</p>

  </div>
</div>

<div class="device-xs visible-xs"></div>
<div class="device-sm visible-sm"></div>
<div class="device-md visible-md"></div>
<div class="device-lg visible-lg"></div>

<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
<script src="../../assets/js/post-jquery.js"></script>
<script src="http://netdna.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
<script src="../../assets/js/post-bootstrap.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  
    ga('create', 'UA-11999401-1', 'auto');
    ga('send', 'pageview');
  </script>
</body>
</html>
