<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="keywords" content="R, Bioconductor, Affymetrix">
  <meta name="author" content="Henrik Bengtsson">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="shortcut icon" type="image/png" href="../../assets/ico/favicon.png">
  <title title="The Aroma Project (Aroma): Block: doCRMAv2() / doASCRMAv2()">Aroma: Block: doCRMAv2() / doASCRMAv2()</title>
  <link href="../../assets/css/bootstrap.min.css" rel="stylesheet">
  <link href="../../assets/css/cbc-tweaks.css" rel="stylesheet">
  <link href="../../assets/css/r_highlight.css" rel="stylesheet">
  <script src="../../assets/js/r_highlight.js"></script>
</head>
<body>
<a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>
<nav class="navbar navbar-default navbar-fixed-top" role="navigation" id="topnavbar">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../../index.html" title="The Aroma Project">Aroma</a>
    </div>

    <div class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../../getstarted/index.html">Get Started</a></li>
        <li><a href="../../docs/index.html">Documentation</a></li>
        <li><a href="../../features/index.html">Features</a></li>
        <li><a href="../../resources/index.html">Resources</a></li>
        <li><a href="../../forum/index.html">Forum</a></li>
        <li><a href="../../FAQ/index.html">FAQ</a></li>
        <li><a href="../../developers/index.html">Developers' Corner</a></li>
        <li><a href="../../about/index.html">About</a></li>
      </ul>
    </div>
  </div>
</nav>

<div id="wrap">
<div class="bs-docs-header" id="content">
<div class="container">
<h1>Block: doCRMAv2() / doASCRMAv2()</h1>

<p>Author: Henrik Bengtsson<br>
 Created on: 2010-05-17<br>
 Last updated on: 2012-06-18</p>

<p>The CRMA v2 <a name=cite-BengtssonH_etal_2009-CRMAv2></a>(<a href="https://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;amp;cmd=prlinks&amp;amp;retmode=ref&amp;amp;id=19535535">Bengtsson, Wirapati, and Speed, 2009</a>) is the recommended
method for estimating full-resolution copy numbers (CN) from all
Affymetrix SNP and CN chip types, including custom-made ones. CRMA v2 can
be done either by doing each step explicitly as in the vignettes, or
using the following <code>doCRMAv2()</code> and <code>doASCRMAv2()</code> methods.</p>

<p>Notes:</p>

<ul>
<li>  If this is your first analysis within the aroma project, please make
sure to first read the &#39;<a href="../../setup/index.html">Setup</a>&#39; and
&#39;<a href="../../definitions/index.html">Definition</a>&#39; pages. This will explain the
importance of following a well defined directory structure and file
names. <em>Understanding this is important and will save you a lot of
time</em>.</li>
<li>  If this is your first analysis with a given chip type, please visit
the &#39;Setup&#39; section of the <a href="../../vignettes/CRMAv2/index.html">CRMA v2 vignette</a> to
make sure that your setup contains all the required annotation
files.</li>
</ul>

<p>Usage:</p>

<pre><code class="r">ds &lt;- doCRMAv2(&quot;HapMap270,testSet&quot;, chipType=&quot;GenomeWideSNP_6&quot;)
ds &lt;- doCRMAv2(&quot;HapMap270,testSet&quot;, chipType=&quot;GenomeWideSNP_6,Full&quot;)
ds &lt;- doCRMAv2(&quot;HapMap270,testSet&quot;, chipType=&quot;GenomeWideSNP_5&quot;)
ds &lt;- doCRMAv2(&quot;HapMap270,testSet&quot;, chipType=&quot;GenomeWideSNP_5,Full,r2&quot;)
ds &lt;- doCRMAv2(&quot;HapMap270,testSet&quot;, chipType=&quot;Mapping250K_Nsp&quot;, plm=&quot;RmaCnPlm&quot;)
ds &lt;- doCRMAv2(&quot;HapMap270,testSet&quot;, chipType=&quot;Mapping50K_Hind240&quot;, plm=&quot;RmaCnPlm&quot;
ds &lt;- doCRMAv2(&quot;GSE8605&quot;, chipType=&quot;Mapping10K_Xba142&quot;, plm=&quot;RmaCnPlm&quot;)
</code></pre>

<p><strong>Argument &#39;verbose&#39;:</strong> Since CRMA v2 will take minutes or hours, we
suggest that you add argument <code>verbose=-10</code> to see some verbose output
while processing the data set.</p>

<p><strong>Argument &#39;plm&#39;:</strong> Note how we specify <code>plm=&quot;RmaCnPlm&quot;</code> for the
10K-500K platforms. We do this in order to control for probe-affinity
effects, which are possible to estimate for those chip types.
Probe-affinity effects are not possible to estimate in the newer chip
types (e.g. GWS) because there all replicated probes are
technical/identical replicates. See the CRMA v2 paper for more details.</p>

<p><strong>Single- vs multi-array method:</strong> When fitting the CRMA v2 model with
<code>plm=&quot;RmaCnPlm&quot;</code>, the method is a multi-array method, whereas with the
default (<code>plm=&quot;AvgCnPlm&quot;</code>), then method is a single-array method. In the
latter case the results for one array will be independent of the other
arrays in the same data set. This make is possible to process a subsets
of the arrays and postponing the others for later and still get the same
results. See the CRMA v2 paper for more details.</p>

<p><strong>Process a subset of the arrays:</strong> In case the single-array method
(<code>plm=&quot;AvgCnPlm&quot;</code>) is used it is possible to process a subset of the
arrays without having to worry about what arrays to including in the
batch. The results will be the same regardless. For instance, the
following two cases will give identical results:</p>

<pre><code class="r">dsA &lt;- doCRMAv2(&quot;HapMap270,testSet&quot;, chipType=&quot;GenomeWideSNP_6&quot;)
dsA &lt;- dsA[c(6,1,3:4)]

dsB &lt;- doCRMAv2(&quot;HapMap270,testSet&quot;, chipType=&quot;GenomeWideSNP_6&quot;, arrays=c(6,1,3:4))
</code></pre>

<p>Note, this is only the case if <code>plm=&quot;AvgCnPlm&quot;</code>.</p>

<p><strong>Allele-specific copy-number (ASCN) estimates:</strong> By adding argument
<code>combineAlleles=FALSE</code> to the above, allele-specific CNs will be
estimated (for SNPs). The default is to estimate total CNs
(<code>combineAlleles=TRUE</code>). Alternatively, use <code>doASCRMAv2()</code>, which is
short for <code>doCRMAv2(..., combineAlleles=FALSE)</code>. The estimated total CNs
will differ slightly when using the two alternatives and the CRMA v2
method has been optimized for total CNs (as in CRMA v2 paper). However,
if you plan to do ASCN analysis, we recommend to use
<code>combineAlleles=FALSE</code> from the beginning, because the statistical
performance is almost as good while you will save lots of time only
processing the data once.</p>

<h2>References</h2>

<p><a
name=bib-BengtssonH_etal_2009-CRMAv2></a><a href="#cite-BengtssonH_etal_2009-CRMAv2">[1]</a>
H. Bengtsson, P. Wirapati, and T. P. Speed. &quot;A single-array
preprocessing method for estimating full-resolution raw copy numbers
from all Affymetrix genotyping arrays including GenomeWideSNP 5 &amp; 6&quot;.
Eng. In: <em>Bioinformatics (Oxford, England)</em> 25.17 (Sep. 2009), pp.
2149-56. ISSN: 1367-4811. DOI:
<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtp371">10.1093/bioinformatics/btp371</a>.
PMID:
<a href="https://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=19535535">19535535</a>.</p>
</div>
</div> 
</div> 

<div id="filler"></div>


<div id="push"></div> 
<div id="footer" style="height: auto;">
  <div class="container">
<p class="muted credit" style="font-size: small;">
  The Aroma Project
  <span class="printonly">(http://www.aroma-project.org/)</span>
  ::
  &copy; Henrik Bengtsson
  ::
  Powered by <a href="http://cran.r-project.org/package=R.rsp">RSP</a>

  <span style="float: right;">
  <a href="http://validator.w3.org/check?st=1;outline=1;verbose=1;group=1&amp;uri=http://www.aroma-project.org/blocks/doCRMAv2/" aria-label="Validate HTML" id="validate-page" target="_blank"><span class="glyphicon glyphicon-check" title="Validate HTML" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Validate HTML</span></a>
  &nbsp;
  <a href="https://github.com/AromaProject/aroma.project.org-website/tree/master/content/blocks/doCRMAv2/index.md.rsp" id="edit"
  target="_blank"><span class="glyphicon glyphicon-edit" title="Edit page" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Edit page</span></a>
  </span>
</p>

  </div>
</div>

<div class="device-xs visible-xs"></div>
<div class="device-sm visible-sm"></div>
<div class="device-md visible-md"></div>
<div class="device-lg visible-lg"></div>

<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
<script src="../../assets/js/post-jquery.js"></script>
<script src="https://netdna.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
<script src="../../assets/js/post-bootstrap.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  
    ga('create', 'UA-11999401-1', 'auto');
    ga('send', 'pageview');
  </script>
</body>
</html>
