<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="keywords" content="R, Bioconductor, Affymetrix">
  <meta name="author" content="Henrik Bengtsson">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="shortcut icon" type="image/png" href="../../assets/ico/favicon.png">
  <title title="The Aroma Project (Aroma): Vignette: Paired total copy number analysis">Aroma: Vignette: Paired total copy number analysis</title>
  <link href="../../assets/css/bootstrap.min.css" rel="stylesheet">
  <link href="../../assets/css/cbc-tweaks.css" rel="stylesheet">
  <link href="../../assets/css/r_highlight.css" rel="stylesheet">
  <script src="../../assets/js/r_highlight.js"></script>
</head>
<body>
<a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>
<nav class="navbar navbar-default navbar-fixed-top" role="navigation" id="topnavbar">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../../index.html" title="The Aroma Project">Aroma</a>
    </div>

    <div class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../../getstarted/index.html">Get Started</a></li>
        <li><a href="../../docs/index.html">Documentation</a></li>
        <li><a href="../../features/index.html">Features</a></li>
        <li><a href="../../resources/index.html">Resources</a></li>
        <li><a href="../../forum/index.html">Forum</a></li>
        <li><a href="../../FAQ/index.html">FAQ</a></li>
        <li><a href="../../developers/index.html">Developers' Corner</a></li>
        <li><a href="../../about/index.html">About</a></li>
      </ul>
    </div>
  </div>
</nav>

<div id="wrap">
<div class="bs-docs-header" id="content">
<div class="container">
<h1>Vignette: Paired total copy number analysis</h1>

<p>Author: Henrik Bengtsson<br>
Created: 2007-09-30<br>
Last updated: 2008-10-27</p>

<p><img src="../../assets/images/CRL-5868D%2Cchr20%2Cx0004.png" alt=""></p>

<p><em>Figure: Raw copy-number estimates and CBS-region estimates on Chromosome 20 for tumor-normal pair (CRL-5868D, CRL-5957D).</em></p>

<div class="alert alert-info" role="alert">
 <p>
  <span class="glyphicon glyphicon-thumbs-up" style="font-size: 1.2em;"></span>
  <em>To help support this work, please consider citing the following relevant references in your publications or talks whenever using their methods or results:</em>
 </p><br>
 <ul>
  <li>
<p>H. Bengtsson, K. Simpson, J. Bullard, et al.
<em>aroma.affymetrix: A generic framework in R for analyzing small to very large Affymetrix data sets in bounded memory</em>.
Tech. rep. 745.
Department of Statistics, University of California, Berkeley, Feb. 2008.</p>

  </li>
  <li>
<p>H. Bengtsson, R. Irizarry, B. Carvalho, et al.
&ldquo;Estimation and assessment of raw copy numbers at the single locus level&rdquo;.
Eng.
In: <em>Bioinformatics (Oxford, England)</em> 24.6 (Mar. 2008), pp. 759-67.
ISSN: 1367-4811.
DOI: <a href="https://doi.org/10.1093/bioinformatics/btn016">10.1093/bioinformatics/btn016</a>.
PMID: <a href="https://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=18204055">18204055</a>.</p>

  </li>
 </ul>
</div>

<h2>Setup</h2>

<h3>Raw data</h3>

<ul>
<li><p>Data set: Affymetrix_2006-TumorNormal</p></li>
<li><p>Tumor-normal pairs: (CRL-2324D, CRL-2325D); (CRL-5868D, CRL-5957D);
(CCL-256D, CCL-256.1D); (CRL-2320D, CRL-2319D); (CRL-2321D, CRL-2362D);
(CRL-2336D, CRL-2337D); (CRL-2338D, CRL-2339D); (CRL-2340D, CRL-2341D);
(CRL-2314D, CRL-2346D)</p></li>
<li><p>Chip types: Mapping250K_Nsp &amp; Mapping250K_Sty</p></li>
</ul>

<p>Source: This data set is publicly available.  Look for public data
sets on Page <a href="../../chipTypes/Mapping250K_Nsp-and-Mapping250K_Sty/index.html">Mapping250K_Nsp &amp;
Mapping250K_Sty</a>.</p>

<p><em>Note</em>: The files downloaded from Affymetrix have &#39;<em>NSP&#39; and &#39;_STY&#39;
parts of their filenames.  <strong>You have to rename the files</strong> by replacing
the &#39;</em>&#39; (underscores) with &#39;,&#39; (commas), e.g. CRL-2324D_NSP.CEL to
CRL-2324D,NSP.CEL and CRL-2324D_STY.CEL to CRL-2324D,STY.CEL.  This way
CEL files for the same sample gets the same <em>name</em> (<a href="../../definitions/namesAndTags/index.html">the part before the
first comma</a>) so they can be automatically
paired, e.g. CRL-2324D.</p>

<p>This is what the rawData/ directory should look like before starting
the analysis:</p>

<pre><code> rawData/  
   Affymetrix_2006-TumorNormal/  
     Mapping250K_Nsp/  
       CRL-2325D,NSP.CEL  
       ...  
       CRL-5957D,NSP.CEL  
    Mapping250K_Sty/  
       CRL-2325D,STY.CEL  
       ...  
       CRL-5957D,STY.CEL
</code></pre>

<h3>Annotation data</h3>

<p>If not already done, you also need to setup the annotationData/
directory for the two chip types.  Please follow the instructions
exactly as given in Vignette <a href="../../vignettes/paired_total_copy_number_analysis/index.html">Total copy number analysis (10K, 100K,
500K)</a>.</p>

<h2>Low-level analysis</h2>

<pre><code class="r">library(&quot;aroma.affymetrix&quot;)
log &lt;- Arguments$getVerbose(-4, timestamp=TRUE)

dataSetName &lt;- &quot;Affymetrix_2006-TumorNormal&quot;
chipTypes &lt;- c(&quot;Mapping250K_Nsp&quot;, &quot;Mapping250K_Sty&quot;)

pairs &lt;- matrix(c(
  &quot;CRL-2325D&quot;, &quot;CRL-2324D&quot;,
  &quot;CRL-5957D&quot;, &quot;CRL-5868D&quot;,
  &quot;CCL-256.1D&quot;, &quot;CCL-256D&quot;,
  &quot;CRL-2319D&quot;, &quot;CRL-2320D&quot;,
  &quot;CRL-2362D&quot;, &quot;CRL-2321D&quot;,
  &quot;CRL-2337D&quot;, &quot;CRL-2336D&quot;,
  &quot;CRL-2339D&quot;, &quot;CRL-2338D&quot;,
  &quot;CRL-2341D&quot;, &quot;CRL-2340D&quot;,
  &quot;CRL-2346D&quot;, &quot;CRL-2314D&quot;
), ncol=2, byrow=TRUE)
colnames(pairs) &lt;- c(&quot;normal&quot;, &quot;tumor&quot;)
</code></pre>

<p>Defining CEL set</p>

<pre><code class="r">csRawList &lt;- list()
for (chipType in chipTypes) {
  cs &lt;- AffymetrixCelSet$byName(dataSetName, chipType=chipType)
  stopifnot(all(getNames(cs) %in% pairs))
  csRawList[[chipType]] &lt;- cs
}

print(csRawList)
</code></pre>

<p>outputs</p>

<pre><code>$Mapping250K_Nsp
AffymetrixCelSet:
Name: Affymetrix_2006-TumorNormal
Tags:
Path: rawData/Affymetrix_2006-TumorNormal/Mapping250K_Nsp
Chip type: Mapping250K_Nsp
Number of arrays: 18
Names: CCL-256.1D, CCL-256D, ..., CRL-5957D
Time period: 2006-01-13 13:07:38 -- 2006-01-16 10:18:56
Total file size: 1127.84MB
RAM: 0.02MB

$Mapping250K_Sty
AffymetrixCelSet:
Name: Affymetrix_2006-TumorNormal
Tags:
Path: rawData/Affymetrix_2006-TumorNormal/Mapping250K_Sty
Chip type: Mapping250K_Sty
Number of arrays: 18
Names: CCL-256.1D, CCL-256D, ..., CRL-5957D
Time period: 2006-01-18 12:57:07 -- 2006-01-18 20:30:57
Total file size: 1127.79MB
RAM: 0.02MB
</code></pre>

<p>Note how the two CEL sets have arrays with matching <em>names</em>, which is
what aroma.affymetrix uses to pair up CEL files from the same
sample.  Note that paired CEL files still can have different <em>tags</em>,
i.e. the filenames may differ as long as the part of the filenames that
define the name is the same for both files.  This is illustrated by the
following code that give details for the first array in each set:</p>

<pre><code class="r">print(csRawList[[&quot;Mapping250K_Nsp&quot;]][[1]])
</code></pre>

<pre><code>AffymetrixCelFile:
Name: **CCL-256.1D**
Tags: **NSP**
Pathname:
rawData/Affymetrix_2006-TumorNormal/Mapping250K_Nsp/**CCL-256.1D,NSP.CEL**
File size: 62.66MB
RAM: 0.01MB
File format: v4 (binary; XDA)
Chip type: Mapping250K_Nsp
Timestamp: 2006-01-16 09:48:51
</code></pre>

<pre><code class="r">print(csRawList[[&quot;Mapping250K_Sty&quot;]][[1]])
</code></pre>

<pre><code>AffymetrixCelFile:
Name: **CCL-256.1D**
Tags: **STY**
Pathname:
rawData/Affymetrix_2006-TumorNormal/Mapping250K_Sty/**CCL-256.1D,STY.CEL**
File size: 62.65MB
RAM: 0.01MB
File format: v4 (binary; XDA)
Chip type: Mapping250K_Sty
Timestamp: 2006-01-18 14:59:36
</code></pre>

<h3>Calibration for allelic crosstalk</h3>

<pre><code class="r">csAccList &lt;- list()
for (chipType in names(csRawList)) {
  cs &lt;- csRawList[[chipType]]
  acc &lt;- AllelicCrosstalkCalibration(cs)
  print(acc)
  csAcc &lt;- process(acc, verbose=log)
  csAccList[[chipType]] &lt;- csAcc
}
</code></pre>

<h3>Summarization</h3>

<pre><code class="r">cesList &lt;- list()
for (chipType in names(csAccList)) {
  cs &lt;- csAccList[[chipType]]
  plm &lt;- RmaCnPlm(cs, mergeStrands=TRUE, combineAlleles=TRUE, shift=+300)
  print(plm)
  fit(plm, verbose=log)
  ces &lt;- getChipEffectSet(plm)
  cesList[[chipType]] &lt;- ces
}
</code></pre>

<h3>PCR fragment length normalization</h3>

<pre><code class="r">cesFlnList &lt;- list()
for (chipType in names(cesList)) {
  ces &lt;- cesList[[chipType]]
  fln &lt;- FragmentLengthNormalization(ces)
  print(fln)
  cesFln &lt;- process(fln, verbose=log)
  cesFlnList[[chipType]] &lt;- cesFln
}
</code></pre>

<p>This gives raw CN estimates for the 18 samples for both chip types.</p>

<h2>Identification of copy-number regions in tumor-normal pairs</h2>

<pre><code class="r"># Split data set in (tumor, normal) pairs
sets &lt;- list(tumor=list(), normal=list())
for (chipType in names(cesFlnList)) {
  ces &lt;- cesFlnList[[chipType]]
  for (type in colnames(pairs)) {
    idxs &lt;- match(pairs[,type], getNames(ces))
    sets[[type]][[chipType]] &lt;- ces[idxs]
  }
}

cns &lt;- CbsModel(sets$tumor, sets$normal)
print(cns)

# Link the ChromosomeExplorer to the segmentation model
ce &lt;- ChromosomeExplorer(cns)
print(ce)

# Fit the model for a few chromosomes
process(ce, chromosomes=c(1, 19, 22), verbose=log);

# The X chromosome is very noisy and generates quite a few missing values
process(ce, chromosomes=23, maxNAFraction=1/5, verbose=log)
</code></pre>
</div>
</div> 
</div> 

<div id="filler"></div>


<div id="push"></div> 
<div id="footer" style="height: auto;">
  <div class="container">
<p class="muted credit" style="font-size: small;">
  The Aroma Project
  <span class="printonly">(https://www.aroma-project.org/)</span>
  ::
  &copy; Henrik Bengtsson
  ::
  Powered by <a href="https://cran.r-project.org/package=R.rsp">RSP</a>

  <span style="float: right;">
  <a href="https://validator.w3.org/check?st=1;outline=1;verbose=1;group=1&amp;uri=https://www.aroma-project.org/vignettes/paired_total_copy_number_analysis/" aria-label="Validate HTML" id="validate-page" target="_blank"><span class="glyphicon glyphicon-check" title="Validate HTML" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Validate HTML</span></a>
  &nbsp;
  <a href="https://github.com/AromaProject/aroma.project.org-website/tree/master/content/vignettes/paired_total_copy_number_analysis/index.md.rsp" id="edit"
  target="_blank"><span class="glyphicon glyphicon-edit" title="Edit page" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Edit page</span></a>
  </span>
</p>

  </div>
</div>

<div class="device-xs visible-xs"></div>
<div class="device-sm visible-sm"></div>
<div class="device-md visible-md"></div>
<div class="device-lg visible-lg"></div>

<script src="../../assets/js/jquery.min.js"></script>
<script src="../../assets/js/post-jquery.js"></script>
<script src="../../assets/js/bootstrap.min.js"></script>
<script src="../../assets/js/post-bootstrap.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  
    ga('create', 'UA-11999401-1', 'auto');
    ga('send', 'pageview');
  </script>
</body>
</html>
