<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="keywords" content="R, Bioconductor, Affymetrix">
  <meta name="author" content="Henrik Bengtsson">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="shortcut icon" type="image/png" href="../../assets/ico/favicon.png">
  <title title="The Aroma Project (Aroma): Vignette: CalMaTe - A calibration method to improve allele-specific copy number of SNP arrays for downstream segmentation">Aroma: Vignette: CalMaTe - A calibration method to improve allele-specific copy number of SNP arrays for downstream segmentation</title>
  <link href="../../assets/css/bootstrap.min.css" rel="stylesheet">
  <link href="../../assets/css/cbc-tweaks.css" rel="stylesheet">
  <link href="../../assets/css/r_highlight.css" rel="stylesheet">
  <script src="../../assets/js/r_highlight.js"></script>
</head>
<body>
<a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>
<nav class="navbar navbar-default navbar-fixed-top" role="navigation" id="topnavbar">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../../index.html" title="The Aroma Project">Aroma</a>
    </div>

    <div class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../../getstarted/index.html">Get Started</a></li>
        <li><a href="../../docs/index.html">Documentation</a></li>
        <li><a href="../../features/index.html">Features</a></li>
        <li><a href="../../resources/index.html">Resources</a></li>
        <li><a href="../../forum/index.html">Forum</a></li>
        <li><a href="../../FAQ/index.html">FAQ</a></li>
        <li><a href="../../developers/index.html">Developers' Corner</a></li>
        <li><a href="../../about/index.html">About</a></li>
      </ul>
    </div>
  </div>
</nav>

<div id="wrap">
<div class="bs-docs-header" id="content">
<div class="container">
<h1>Vignette: CalMaTe - A calibration method to improve allele-specific copy number of SNP arrays for downstream segmentation</h1>

<p>Created on: 2010-07-30<br>
Last updated: 2014-12-21</p>

<h2>Introduction</h2>

<p><em>CalMaTe</em> calibrates preprocessed allele-specific copy-number estimates (ASCNs) from DNA microarrays by controlling for SNP-specific allelic crosstalk. The resulting ASCNs are on average more accurate, which increases the power of segmentation methods (e.g. PSCBS by <a name=cite-OlshenA_etal_2011-PSCBS></a><a href="https://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;amp;cmd=prlinks&amp;amp;retmode=ref&amp;amp;id=21666266">Olshen, Bengtsson, Neuvial, Spellman, Olshen, and Seshan (2011)</a>) for detecting changes between copy-number states in tumor studies including copy-neutral loss of heterozygosity (LOH).  CalMaTe applies to any ASCNs regardless of preprocessing method and microarray technology, e.g. Affymetrix and Illumina.
Contrary to the TumorBoost <a name=cite-BengtssonH_etal_2010-TumorBoost></a>(<a href="https://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;amp;cmd=prlinks&amp;amp;retmode=ref&amp;amp;id=20462408">Bengtsson, Neuvial, and Speed, 2010</a>), which requires a single pair of tumor-normal samples, the CalMaTe methods don&#39;t require matched normals, but instead it requires a large number of reference samples.
For more details, see <a name=cite-OrtizM_etal_2012-CalMaTe></a><a href="https://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;amp;cmd=prlinks&amp;amp;retmode=ref&amp;amp;id=22576175">Ortiz-Estevez, Aramburu, Bengtsson, Neuvial, and Rubio (2012)</a>.</p>

<h2>Data</h2>

<p>Below we will use the public GEO data set <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE12702">GSE12702</a>:</p>

<pre><code>rawData/  
   GSE12702/  
     Mapping250K_Nsp/  
       GSM318728.CEL, GSM318729.CEL, ..., GSM318767.CEL
</code></pre>

<h2>Analysis</h2>

<h3>Allele-specific copy-number estimation (CRMA v2)</h3>

<p>We will use an allele-specific version of the CRMA v2 method <a name=cite-BengtssonH_etal_2009-CRMAv2></a>(<a href="https://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;amp;cmd=prlinks&amp;amp;retmode=ref&amp;amp;id=19535535">Bengtsson, Wirapati, and Speed, 2009</a>) to estimate allele-specific copy numbers (ASCNs) at each SNP.   The CRMA v2 method is implemented in the <a href="http://cran.r-project.org/package=aroma.affymetrix">aroma.affymetrix</a> package and the CalMaTe method in the <a href="http://cran.r-project.org/package=calmate">calmate</a> package.</p>

<pre><code class="r">library(&quot;aroma.affymetrix&quot;)
library(&quot;calmate&quot;)
verbose &lt;- Arguments$getVerbose(-8, timestamp=TRUE)

# Setting up data set
csR &lt;- AffymetrixCelSet$byName(&quot;GSE12702&quot;, chipType=&quot;Mapping250K_Nsp&quot;)
print(csR)

## AffymetrixCelSet:
## Name: GSE12702
## Tags:
## Path: rawData/GSE12702/Mapping250K_Nsp
## Platform: Affymetrix
## Chip type: Mapping250K_Nsp
## Number of arrays: 40
## Names: GSM318728, GSM318729, ..., GSM318767
## Time period: 2008-04-03 13:52:53 -- 2008-04-10 20:08:49
## Total file size: 2506.11MB
## RAM: 0.06MB


# Allele-specific CRMA v2 with log-additive modeling
dsList &lt;- doASCRMAv2(csR, plm=&quot;RmaCnPlm&quot;, verbose=verbose)
print(dsList)

## $total
## AromaUnitTotalCnBinarySet:
## Name: GSE12702
## Tags: ACC,-XY,BPN,-XY,RMA,FLN,-XY
## Full name: GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY
## Number of files: 40
## Names: GSM318728, GSM318729, ..., GSM318767 [40]
## Path (to the first file): totalAndFracBData/GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY/Mapping250K_Nsp
## Total file size: 40.05 MB
## RAM: 0.06MB##
## $fracB
## AromaUnitFracBCnBinarySet:
## Name: GSE12702
## Tags: ACC,-XY,BPN,-XY,RMA,FLN,-XY
## Full name: GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY
## Number of files: 40
## Names: GSM318728, GSM318729, ..., GSM318767 [40]
## Path (to the first file): totalAndFracBData/GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY/Mapping250K_Nsp
## Total file size: 40.05 MB
## RAM: 0.05MB
</code></pre>

<h3>Calibration of ASCN estimates (CalMaTe)</h3>

<pre><code class="r">cmt &lt;- CalMaTeCalibration(dsList)
print(cmt)

## CalMaTeCalibration:
## Data sets (2):
## &lt;Total&gt;:
## AromaUnitTotalCnBinarySet:
## Name: GSE12702
## Tags: ACC,-XY,BPN,-XY,RMA,FLN,-XY
## Full name: GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY
## Number of files: 40
## Names: GSM318728, GSM318729, ..., GSM318767 [40]
## Path (to the first file): totalAndFracBData/GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY/Mapping250K_Nsp
## Total file size: 40.05 MB
## RAM: 0.06MB
## &lt;FracB&gt;:
## AromaUnitFracBCnBinarySet:
## Name: GSE12702
## Tags: ACC,-XY,BPN,-XY,RMA,FLN,-XY
## Full name: GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY
## Number of files: 40
## Names: GSM318728, GSM318729, ..., GSM318767 [40]
## Path (to the first file): totalAndFracBData/GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY/Mapping250K_Nsp
## Total file size: 40.05 MB
## RAM: 0.05MB

dsCList &lt;- process(cmt, verbose=verbose)
print(dsCList)

## $total
## AromaUnitTotalCnBinarySet:
## Name: GSE12702
## Tags: ACC,-XY,BPN,-XY,RMA,FLN,-XY,CMTN
## Full name: GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY,CMTN
## Number of files: 40
## Names: GSM318728, GSM318729, ..., GSM318767 [40]
## Path (to the first file): totalAndFracBData/GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY,CMTN/Mapping250K_Nsp
## Total file size: 40.05 MB
## RAM: 0.06MB
##
## $fracB
## AromaUnitFracBCnBinarySet:
## Name: GSE12702
## Tags: ACC,-XY,BPN,-XY,RMA,FLN,-XY,CMTN
## Full name: GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY,CMTN
## Number of files: 40
## Names: GSM318728, GSM318729, ..., GSM318767 [40]
## Path (to the first file): totalAndFracBData/GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY,CMTN/Mapping250K_Nsp
## Total file size: 40.05 MB
## RAM: 0.05MB
</code></pre>

<h4>CalMaTe calibration using only a subset of the data set as references</h4>

<p>As this particular data sets consists of tumor and normal samples, it
may be advantageous to use only normal samples for the calibration
step (and then backtransform all arrays using the estimated
calibration).  First we identify the indices of the samples that should
be used as references:</p>

<pre><code class="r">names &lt;- getNames(dsList$total)

patientIDs &lt;- c(24, 25, 27, 31, 45, 52, 58, 60, 75, 110, 115, 122, 128, 137, 138, 140, 154, 167, 80, 96)
sampleTypes &lt;- c(&quot;tumor&quot;, &quot;normal&quot;)

pids &lt;- rep(patientIDs, each=length(sampleTypes))
types &lt;- rep(sampleTypes, times=length(patientIDs))

mat &lt;- cbind(names, pids, types)
str(mat)
## chr [1:40, 1:3] &quot;GSM318728&quot; &quot;GSM318729&quot; &quot;GSM318730&quot; &quot;GSM318731&quot; ...
##  - attr(*, &quot;dimnames&quot;)=List of 2
##   ..$ : NULL
##   ..$ : chr [1:3] &quot;names&quot; &quot;pids&quot; &quot;types&quot;

idxsN &lt;- which(mat[, &quot;types&quot;] == &quot;normal&quot;)
</code></pre>

<p>Next, when setting up the CalMaTe model, we specify the reference
samples using argument <code>references</code> as follows:</p>

<pre><code class="r">cmtN &lt;- CalMaTeCalibration(dsList, tags=c(&quot;*&quot;, &quot;normalReferences&quot;), references=idxsN)
print(cmtN)

## CalMaTeCalibration:
##   Data sets (2):
##   &lt;Total&gt;:
##   AromaUnitTotalCnBinarySet:
##   Name: GSE12702
## Tags: ACC,-XY,BPN,-XY,RMA,FLN,-XY
## Full name: GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY
## Number of files: 40
## Names: GSM318728, GSM318729, ..., GSM318767 [40]
## Path (to the first file): totalAndFracBData/GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY/Mapping250K_Nsp
## Total file size: 40.05 MB
## RAM: 0.06MB
## &lt;FracB&gt;:
##   AromaUnitFracBCnBinarySet:
##   Name: GSE12702
## Tags: ACC,-XY,BPN,-XY,RMA,FLN,-XY
## Full name: GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY
## Number of files: 40
## Names: GSM318728, GSM318729, ..., GSM318767 [40]
## Path (to the first file): totalAndFracBData/GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY/Mapping250K_Nsp
## Total file size: 40.05 MB
## RAM: 0.05MB

dsCNList &lt;- process(cmtN, verbose=verbose)
print(dsCNList)

## $total
## AromaUnitTotalCnBinarySet:
## Name: GSE12702
## Tags: ACC,-XY,BPN,-XY,RMA,FLN,-XY,CMTN,normalReferences
## Full name: GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY,CMTN,normalReferences
## Number of files: 40
## Names: GSM318728, GSM318729, ..., GSM318767 [40]
## Path (to the first file): totalAndFracBData/GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY,CMTN,normalReferences/Mapping250K_Nsp
## Total file size: 40.05 MB
## RAM: 0.06MB

## $fracB
## AromaUnitFracBCnBinarySet:
## Name: GSE12702
## Tags: ACC,-XY,BPN,-XY,RMA,FLN,-XY,CMTN,normalReferences
## Full name: GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY,CMTN,normalReferences
## Number of files: 40
## Names: GSM318728, GSM318729, ..., GSM318767 [40]
## Path (to the first file): totalAndFracBData/GSE12702,ACC,-XY,BPN,-XY,RMA,FLN,-XY,CMTN,normalReferences/Mapping250K_Nsp
## Total file size: 40.05
## RAM: 0.05MB
</code></pre>

<h3>Results</h3>

<pre><code class="r">extractSignals &lt;- function(dsList, sampleName, reference=c(&quot;none&quot;, &quot;median&quot;), refIdxs=NULL, ..., verbose=FALSE) {
    reference &lt;- match.arg(reference)
      idx &lt;- indexOf(dsList$total, sampleName)
      dfT &lt;- dsList$total[[idx]]
      dfB &lt;- dsList$fracB[[idx]]
      tcn &lt;- extractRawCopyNumbers(dfT, logBase=NULL, ..., verbose=verbose)
      baf &lt;- extractRawAlleleBFractions(dfB, ..., verbose=verbose)
      if (reference == &quot;median&quot;) {
        if (!is.null(refIdxs)) {
          dsR &lt;- dsList$total[refIdxs]
        } else {
          dsR &lt;- dsList$total
        }
        dfTR &lt;- getAverageFile(dsR, verbose=verbose)
        tcnR &lt;- extractRawCopyNumbers(dfTR, logBase=NULL, ..., verbose=verbose)
        tcn &lt;- divideBy(tcn, tcnR)
        setSignals(tcn, 2*getSignals(tcn))
      }
    list(tcn=tcn, baf=baf)
} # extractSignals()

sampleName &lt;- &quot;GSM318736&quot;

pch &lt;- 19
cex &lt;- 0.8

snT &lt;- sampleName
chr &lt;- 8

for (normalRefs in c(TRUE, FALSE)) {
  if (normalRefs) {
    figName &lt;- sprintf(&quot;%s,Chr%02d,CalMaTe,normalReferences&quot;, snT, chr)
    dataT &lt;- extractSignals(dsList, sampleName=snT, chromosome=chr, reference=&quot;median&quot;, refIdxs=idxsN, verbose=verbose)
    dataTC &lt;- extractSignals(dsCNList, sampleName=snT, chromosome=chr, verbose=verbose)
  } else {
    figName &lt;- sprintf(&quot;%s,Chr%02d,CalMaTe&quot;, snT, chr)
    dataT &lt;- extractSignals(dsList, sampleName=snT, chromosome=chr, reference=&quot;median&quot;, verbose=verbose)
    dataTC &lt;- extractSignals(dsCList, sampleName=snT, chromosome=chr, verbose=verbose)
  }


  toPNG(figName, width=1200, {
    subplots(4, ncol=1)
    par(mar=c(2,5,1,1)+0.1, cex=cex, cex.lab=2.4, cex.axis=2.2)

    plot(dataT$tcn, ylim=c(0,4), pch=pch)
    plot(dataT$baf, pch=pch)
    plot(dataTC$tcn, ylim=c(0,4), pch=pch)
    plot(dataTC$baf, pch=pch)
  })
}
</code></pre>

<p><img src="../../assets/images/GSM318736%2CChr08%2CCalMaTe.png" alt="TCN and BAF before and after CalMaTe"></p>

<p><em>Figure 1a: TCNs and BAFs along chromosome 8 of sample GSM318736 with and
without CalMaTe when using all 40 samples are used for calibration.
Data is from CRMA v2-processed Affymetrix Mapping250K_Nsp arrays.
Top two panels show the TCNs and BAFs before CalMaTe.
Bottom two panels show the TCNs and BAFs after CalMaTe.
The non-calibrated BAFs are noisy and biased
(homozygous clouds are notat 0 and 1) whereas the CalMaTe-calibrated
BAFs are much less noisy and located closer to the expected locations.
We find that there are three regions, a normal region at 0-20Mb with 2
copies and BAFs near 0, 1/2 and 1 (corresponding to AA, AB and BB
genotypes), a deletion at 20-44Mb with1 copy and BAFs close to 0 and 1
(A and B), and a gain at 49-145Mb with 3 copies and BAFs near 0, 1/3,
2/3, and 1 (AAA, AAB, ABB and BBB). The BAFs in the deleted region are
noisier because of weaker ASCNs.</em></p>

<p><img src="../../assets/images/GSM318736%2CChr08%2CCalMaTe%2CnormalReferences.png" alt="TCN and BAF before and after CalMaTe using normals as references"></p>

<p><em>Figure 1b: TCNs and BAFs along chromosome 8 of sample GSM318736 with and
without CalMaTe when using all 40 samples are used for calibration.
Same data as in Figure 1a.
Top two panels show the TCNs and BAFs before CalMaTe.
Bottom two panels show the TCNs and BAFs after CalMaTe.</em></p>

<h3>(betaN, betaT) plots</h3>

<p>Another way to visualize the effect of CalMaTe on allelic signals, and
connect/contrast it with TumorBoost, is to look at tumor-normal pairs
and compare allele B fraction in the tumor to allele B fraction in the
normal, before and after CalMaTe and/or TumorBoost.</p>

<pre><code class="r">mm &lt;- match(sampleName, names)
id &lt;- pid[mm]

idxT &lt;- which((pid==id) &amp; (type==&quot;tumor&quot;))
snT &lt;- names[idxT]
stopifnot(snT==sampleName)

idxN &lt;- which((pid==id) &amp; (type==&quot;normal&quot;))
snN &lt;- names[idxN]

xlab &lt;- &quot;Normal BAF&quot;
ylab &lt;- &quot;TumorBAF&quot;
lim &lt;- c(-0.1, 1.1)
cext &lt;- 1.8

regions &lt;- list(&quot;normal (1,1)&quot;=c(0, 16), &quot;loss (0,1)&quot;=c(16.5, 45), &quot;gain (1,2)&quot;=c(45, 150))

for (rr in seq(along=regions)) {
  reg &lt;- regions[[rr]]
  regLab &lt;- paste(reg, collapse=&quot;-&quot;)
  cnLab &lt;- names(regions)[rr]
  datT &lt;- extractSignals(dsList, sampleName=snT, chromosome=chr, reg=reg*1e6, verbose=verbose)
  datN &lt;- extractSignals(dsList, sampleName=snN, chromosome=chr, reg=reg*1e6, verbose=verbose)

  datCT &lt;- extractSignals(dsCList, sampleName=snT, chromosome=chr, reg=reg*1e6, verbose=verbose)
  datCN &lt;- extractSignals(dsCList, sampleName=snN, chromosome=chr, reg=reg*1e6, verbose=verbose)

  tbn &lt;- normalizeTumorBoost(datT$baf$y, datN$baf$y, verbose=verbose)
  tbnC &lt;- normalizeTumorBoost(datCT$baf$y, datCN$baf$y, verbose=verbose)

  figName &lt;- sprintf(&quot;%s,betaNvsBetaT,Chr%02d,%s&quot;, snT, chr, regLab)
  toPNG(figName, width=1200, {
    subplots(4, ncol=2)
    par(mar=c(5,5,2,1)+0.1, cex=cex, cex.lab=2.4, cex.axis=2.2)
    plot(datN$baf$y, datT$baf$y, pch=pch, xlim=lim, ylim=lim, xlab=xlab, ylab=ylab)
    stext(&quot;ASCRMAv2&quot;, side=3, pos=1, cex=cext)
    stext(cnLab, side=3, pos=0, cex=cext)
    plot(datCN$baf$y, datCT$baf$y, pch=pch, xlim=lim, ylim=lim, xlab=xlab, ylab=ylab)
    stext(&quot;ASCRMAv2 + CalMaTe&quot;, side=3, pos=1, cex=cext)
    stext(cnLab, side=3, pos=0, cex=cext)
    plot(datN$baf$y, tbn, pch=pch, xlim=lim, ylim=lim, xlab=xlab, ylab=ylab)
    stext(&quot;ASCRMAv2 + TumorBoost&quot;, side=3, pos=1, cex=cext)
    stext(cnLab, side=3, pos=0, cex=cext)
    plot(datCN$baf$y, tbnC, pch=pch, xlim=lim, ylim=lim, xlab=xlab, ylab=ylab)
    stext(&quot;ASCRMAv2 + CalMaTe + TumorBoost&quot;, side=3, pos=1, cex=cext)
    stext(cnLab, side=3, pos=0, cex=cext)
  })
}
</code></pre>

<p>In Figures 2-4, paired tumor-normal BAFs for three (manually identified) copy number regions on chromosome 8 of sample GSM318736 are displayed for each of the following four combinations of preprocessing methods: ASCRMA v2 (top left),
ASCRMA v2 + CalMaTe (top right), ASCRMA v2 + TumorBoost (bottom left),
ASCRMA v2 + CalMaTe + TumorBoost (bottom right).</p>

<p><img src="../../assets/images/GSM318736%2CbetaNvsBetaT%2CChr08%2C0-16.png" alt="betaT vs betaN in a normal region"></p>

<p><em>Figure 2: betaT vs betaN in a normal (1,1) region between 0 and 16 Mb on Chromosome 8.  Preprocessing methods used: ASCRMA v2 (top left), ASCRMA v2 + CalMaTe (top right), ASCRMA v2 + TumorBoost (bottom left), ASCRMA v2 + CalMaTe + TumorBoost (bottom right).</em></p>

<p><img src="../../assets/images/GSM318736%2CbetaNvsBetaT%2CChr08%2C16.5-45.png" alt="betaT vs betaN in a region of homozygous deletion
(0,1)"></p>

<p><em>Figure 3: betaT vs betaN in a region of homozygous deletion (0,1) between 16 and 45 Mb on Chromosome 8.  Preprocessing methods used: ASCRMA v2 (top left), ASCRMA v2 + CalMaTe (top right), ASCRMA v2 + TumorBoost (bottom left), ASCRMA v2 + CalMaTe + TumorBoost (bottom right).</em></p>

<p><img src="../../assets/images/GSM318736%2CbetaNvsBetaT%2CChr08%2C45-150.png" alt="betaT vs betaN in a region of gain
(1,2)"></p>

<p><em>Figure 4: betaT vs betaN in a region of gain (1,2) after 45 Mb on Chromosome 8.  Preprocessing methods used: ASCRMA v2 (top left), ASCRMA v2 + CalMaTe (top right), ASCRMA v2 + TumorBoost (bottom left), ASCRMA v2 + CalMaTe + TumorBoost (bottom right).</em></p>

<p>Note: when comparing signals from TumorBoost to CalMaTe, it is useful
to keep in mind that <strong>TumorBoost does not normalize the normal
sample</strong>, whereas CalMaTe calibrates all samples regardless of their
types.  This means that only the y axes are comparable across the 4
plots.  Also, one should focus on SNPs that are heterozygous in the
normal samples (they correspond to normal BAFs close to 1/2) as in
theory, only these SNPs are informative in terms of copy number changes.</p>

<h2>References</h2>

<p><a
name=bib-BengtssonH_etal_2009-CRMAv2></a><a href="#cite-BengtssonH_etal_2009-CRMAv2">[1]</a>
H. Bengtsson, P. Wirapati and T. P. Speed. &quot;A single-array
preprocessing method for estimating full-resolution raw copy numbers
from all Affymetrix genotyping arrays including GenomeWideSNP 5 &amp; 6&quot;.
Eng. In: <em>Bioinformatics (Oxford, England)</em> 25.17 (Sep. 2009), pp.
2149-56. ISSN: 1367-4811. DOI:
<a href="http://dx.doi.org/10.1093/bioinformatics/btp371">10.1093/bioinformatics/btp371</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=19535535">19535535</a>.</p>

<p><a
name=bib-BengtssonH_etal_2010-TumorBoost></a><a href="#cite-BengtssonH_etal_2010-TumorBoost">[2]</a>
H. Bengtsson, P. Neuvial and T. P. Speed. &quot;TumorBoost: normalization of
allele-specific tumor copy numbers from a single pair of tumor-normal
genotyping microarrays&quot;. Eng. In: <em>BMC bioinformatics</em> 11 (2010), p.
245. ISSN: 1471-2105. DOI:
<a href="http://dx.doi.org/10.1186/1471-2105-11-245">10.1186/1471-2105-11-245</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=20462408">20462408</a>.</p>

<p><a
name=bib-OlshenA_etal_2011-PSCBS></a><a href="#cite-OlshenA_etal_2011-PSCBS">[3]</a>
A. B. Olshen, H. Bengtsson, P. Neuvial, et al. &quot;Parent-specific copy
number in paired tumor-normal studies using circular binary
segmentation&quot;. Eng. In: <em>Bioinformatics (Oxford, England)</em> 27.15 (Aug.
2011), pp. 2038-46. ISSN: 1367-4811. DOI:
<a href="http://dx.doi.org/10.1093/bioinformatics/btr329">10.1093/bioinformatics/btr329</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=21666266">21666266</a>.</p>

<p><a
name=bib-OrtizM_etal_2012-CalMaTe></a><a href="#cite-OrtizM_etal_2012-CalMaTe">[4]</a>
M. Ortiz-Estevez, A. Aramburu, H. Bengtsson, et al. &quot;CalMaTe: a method
and software to improve allele-specific copy number of SNP arrays for
downstream segmentation&quot;. Eng. In: <em>Bioinformatics (Oxford, England)</em>
28.13 (Jul. 2012), pp. 1793-4. ISSN: 1367-4811. DOI:
<a href="http://dx.doi.org/10.1093/bioinformatics/bts248">10.1093/bioinformatics/bts248</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=22576175">22576175</a>.</p>
</div>
</div> 
</div> 

<div id="filler"></div>


<div id="push"></div> 
<div id="footer" style="height: auto;">
  <div class="container">
<p class="muted credit" style="font-size: small;">
  The Aroma Project
  <span class="printonly">(http://www.aroma-project.org/)</span>
  ::
  &copy; Henrik Bengtsson
  ::
  Powered by <a href="http://cran.r-project.org/package=R.rsp">RSP</a>

  <span style="float: right;">
  <a href="http://validator.w3.org/check?st=1;outline=1;verbose=1;group=1&amp;uri=http://www.aroma-project.org/vignettes/CalMaTe/" aria-label="Validate HTML" id="validate-page" target="_blank"><span class="glyphicon glyphicon-check" title="Validate HTML" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Validate HTML</span></a>
  &nbsp;
  <a href="https://github.com/AromaProject/aroma.project.org-website/tree/master/content/vignettes/CalMaTe/index.md.rsp" id="edit"
  target="_blank"><span class="glyphicon glyphicon-edit" title="Edit page" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Edit page</span></a>
  </span>
</p>

  </div>
</div>

<div class="device-xs visible-xs"></div>
<div class="device-sm visible-sm"></div>
<div class="device-md visible-md"></div>
<div class="device-lg visible-lg"></div>

<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
<script src="../../assets/js/post-jquery.js"></script>
<script src="http://netdna.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
<script src="../../assets/js/post-bootstrap.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  
    ga('create', 'UA-11999401-1', 'auto');
    ga('send', 'pageview');
  </script>
</body>
</html>
