<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="keywords" content="R, Bioconductor, Affymetrix">
  <meta name="author" content="Henrik Bengtsson">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="shortcut icon" type="image/png" href="../assets/ico/favicon.png">
  <title title="The Aroma Project (Aroma): Publications">Aroma: Publications</title>
  <link href="../assets/css/bootstrap.min.css" rel="stylesheet">
  <link href="../assets/css/cbc-tweaks.css" rel="stylesheet">
  <link href="../assets/css/r_highlight.css" rel="stylesheet">
  <script src="../assets/js/r_highlight.js"></script>
</head>
<body>
<a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>
<nav class="navbar navbar-default navbar-fixed-top" role="navigation" id="topnavbar">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html" title="The Aroma Project">Aroma</a>
    </div>

    <div class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="../getstarted/index.html">Get Started</a></li>
        <li><a href="../docs/index.html">Documentation</a></li>
        <li><a href="../features/index.html">Features</a></li>
        <li><a href="../resources/index.html">Resources</a></li>
        <li><a href="../forum/index.html">Forum</a></li>
        <li><a href="../FAQ/index.html">FAQ</a></li>
        <li><a href="../developers/index.html">Developers' Corner</a></li>
        <li><a href="../about/index.html">About</a></li>
      </ul>
    </div>
  </div>
</nav>

<div id="wrap">
<div class="bs-docs-header" id="content">
<div class="container">
<h1>Publications</h1>

<p>[1] M. Ortiz-Estevez, A. Aramburu, H. Bengtsson, et al. &quot;CalMaTe: a
method and software to improve allele-specific copy number of SNP
arrays for downstream segmentation&quot;. Eng. In: <em>Bioinformatics (Oxford,
England)</em> 28.13 (2012), pp. 1793-4. DOI:
<a href="http://dx.doi.org/10.1093/bioinformatics/bts248">10.1093/bioinformatics/bts248</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=22576175">22576175</a>.<br>
Keywords: CalMaTe; multi-array method</p>

<p>[2] P. Neuvial, H. Bengtsson and T. P. Speed. &quot;Statistical Analysis of
Single Nucleotide Polymorphism Microarrays in Cancer Studies&quot;. In:
<em>Handbook of Statistical Bioinformatics</em>. Springer Berlin Heidelberg,
2011, pp. 225-255. DOI:
<a href="http://dx.doi.org/10.1007/978-3-642-16345-6_11">10.1007/978-3-642-16345-6_11</a>.
URL:
<a href="http://dx.doi.org/10.1007/978-3-642-16345-6_11">http://dx.doi.org/10.1007/978-3-642-16345-6_11</a>.</p>

<p>[3] A. B. Olshen, H. Bengtsson, P. Neuvial, et al. &quot;Parent-specific
copy number in paired tumor-normal studies using circular binary
segmentation&quot;. Eng. In: <em>Bioinformatics (Oxford, England)</em> 27.15
(2011), pp. 2038-46. DOI:
<a href="http://dx.doi.org/10.1093/bioinformatics/btr329">10.1093/bioinformatics/btr329</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=21666266">21666266</a>.<br>
Keywords: PSCBS; parent-specific copy numbers; segmentation; tumor-normal samples; paired; non-paired; single-patient method</p>

<p>[4] H. Bengtsson, P. Neuvial and T. P. Speed. &quot;TumorBoost:
normalization of allele-specific tumor copy numbers from a single pair
of tumor-normal genotyping microarrays&quot;. Eng. In: <em>BMC bioinformatics</em>
11 (2010), p. 245. DOI:
<a href="http://dx.doi.org/10.1186/1471-2105-11-245">10.1186/1471-2105-11-245</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=20462408">20462408</a>.<br>
Keywords: TumorBoost; paired tumor-normal samples; single-patient method</p>

<p>[5] M. Ortiz-Estevez, H. Bengtsson and A. Rubio. &quot;ACNE: a summarization
method to estimate allele-specific copy numbers for Affymetrix SNP
arrays&quot;. Eng. In: <em>Bioinformatics (Oxford, England)</em> 26.15 (2010), pp.
1827-33. DOI:
<a href="http://dx.doi.org/10.1093/bioinformatics/btq300">10.1093/bioinformatics/btq300</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=20529889">20529889</a>.<br>
Keywords: ACNE; multi-array method</p>

<p>[6] H. Bengtsson, A. Ray, P. Spellman, et al. &quot;A single-sample method
for normalizing and combining full-resolution copy numbers from
multiple platforms, labs and analysis methods&quot;. Eng. In:
<em>Bioinformatics (Oxford, England)</em> 25.7 (2009), pp. 861-7. DOI:
<a href="http://dx.doi.org/10.1093/bioinformatics/btp074">10.1093/bioinformatics/btp074</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=19193730">19193730</a>.<br>
Keywords: MSCN; multi-source copy-number normalization; single-patient method</p>

<p>[7] <a
name=bib-BengtssonH_etal_2009-CRMAv2></a><a href="#cite-BengtssonH_etal_2009-CRMAv2">[1]</a>
H. Bengtsson, P. Wirapati and T. P. Speed. &quot;A single-array
preprocessing method for estimating full-resolution raw copy numbers
from all Affymetrix genotyping arrays including GenomeWideSNP 5 &amp; 6&quot;.
Eng. In: <em>Bioinformatics (Oxford, England)</em> 25.17 (2009), pp. 2149-56.
DOI:
<a href="http://dx.doi.org/10.1093/bioinformatics/btp371">10.1093/bioinformatics/btp371</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=19535535">19535535</a>.<br>
Keywords: CRMA v2; single-array method</p>

<p>[8] M. D. Robinson and T. P. Speed. &quot;Differential splicing using
whole-transcript microarrays&quot;. Eng. In: <em>BMC bioinformatics</em> 10 (2009),
p. 156. DOI:
<a href="http://dx.doi.org/10.1186/1471-2105-10-156">10.1186/1471-2105-10-156</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=19463149">19463149</a>.<br>
Keywords: FIRMAGene; multi-array method</p>

<p>[9] H. Bengtsson, R. Irizarry, B. Carvalho, et al. &quot;Estimation and
assessment of raw copy numbers at the single locus level&quot;. Eng. In:
<em>Bioinformatics (Oxford, England)</em> 24.6 (2008), pp. 759-67. DOI:
<a href="http://dx.doi.org/10.1093/bioinformatics/btn016">10.1093/bioinformatics/btn016</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=18204055">18204055</a>.<br>
Keywords: CRMA v1; multi-array method</p>

<p>[10] H. Bengtsson, K. Simpson, J. Bullard, et al. <em>aroma.affymetrix: A
generic framework in R for analyzing small to very large Affymetrix
data sets in bounded memory</em>. Tech. rep. 745. Department of Statistics,
University of California, Berkeley, Feb. 2008.<br>
Keywords: aroma.affymetrix; preprocessing; normalization; quality control; background correction; probe-sequence normalization; quantile normalization; probeset summarization</p>

<p>[11] E. Purdom, K. M. Simpson, M. D. Robinson, et al. &quot;FIRMA: a method
for detection of alternative splicing from exon array data&quot;. Eng. In:
<em>Bioinformatics (Oxford, England)</em> 24.15 (2008), pp. 1707-14. DOI:
<a href="http://dx.doi.org/10.1093/bioinformatics/btn284">10.1093/bioinformatics/btn284</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=18573797">18573797</a>.<br>
Keywords: FIRMA; gene expressions; exon-specific expressions; multi-array method</p>

<p>[12] B. Carvalho, H. Bengtsson, T. P. Speed, et al. &quot;Exploration,
normalization, and genotype calls of high-density oligonucleotide SNP
array data&quot;. Eng. In: <em>Biostatistics (Oxford, England)</em> 8.2 (2007), pp.
485-99. DOI:
<a href="http://dx.doi.org/10.1093/biostatistics/kxl042">10.1093/biostatistics/kxl042</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=17189563">17189563</a>.<br>
Keywords: CRLMM; genotyping; multi-array method</p>

<p>[13] M. D. Robinson and T. P. Speed. &quot;A comparison of Affymetrix gene
expression arrays&quot;. Eng. In: <em>BMC bioinformatics</em> 8 (2007), p. 449.
DOI:
<a href="http://dx.doi.org/10.1186/1471-2105-8-449">10.1186/1471-2105-8-449</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=18005448">18005448</a>.</p>

<p>[14] A. Bengtsson and H. Bengtsson. &quot;Microarray image analysis:
background estimation using quantile and morphological filters&quot;. Eng.
In: <em>BMC bioinformatics</em> 7 (2006), p. 96. DOI:
<a href="http://dx.doi.org/10.1186/1471-2105-7-96">10.1186/1471-2105-7-96</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=16504173">16504173</a>.<br>
Keywords: image analysis; background correction; single-array method</p>

<p>[15] H. Bengtsson and O. Hössjer. &quot;Methodological study of affine
transformations of gene expression data with proposed robust
non-parametric multi-dimensional normalization method&quot;. Eng. In: <em>BMC
bioinformatics</em> 7 (2006), p. 100. DOI:
<a href="http://dx.doi.org/10.1186/1471-2105-7-100">10.1186/1471-2105-7-100</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=16509971">16509971</a>.<br>
Keywords: affine normalization; single-array method</p>

<p>[16] H. Bengtsson, G. Jönsson and J. Vallon-Christersson. &quot;Calibration
and assessment of channel-specific biases in microarray data with
extended dynamical range&quot;. Eng. In: <em>BMC bioinformatics</em> 5 (2004), p.
177. DOI:
<a href="http://dx.doi.org/10.1186/1471-2105-5-177">10.1186/1471-2105-5-177</a>.
PMID:
<a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;cmd=prlinks&amp;retmode=ref&amp;id=15541170">15541170</a>.<br>
Keywords: multiscan calibration; single-array method</p>

<p>[17] H. Bengtsson. &quot;The R.oo package - Object-Oriented Programming with
References Using Standard R Code&quot;. In: <em>Proceedings of the 3rd
International Workshop on Distributed Statistical Computing (DSC
2003)</em>. Ed. by K. Hornik, F. Leisch and A. Zeileis. Vienna, Austria,
2003. URL:
<a href="http://www.ci.tuwien.ac.at/Conferences/DSC-2003/">http://www.ci.tuwien.ac.at/Conferences/DSC-2003/</a>.<br>
Keywords: R.methodsS3; R.oo; S3</p>

<p>[18] H. Bengtsson. <em>Identification and normalization of plate effects in
cDNA microarray data</em>. Preprints in Mathematical Sciences 2002:28.
MatStatLU, 2002.<br>
Keywords: print-tip normalization; print-order effects; plate-order effects</p>
</div>
</div> 
</div> 

<div id="filler"></div>


<div id="push"></div> 
<div id="footer" style="height: auto;">
  <div class="container">
<p class="muted credit" style="font-size: small;">
  The Aroma Project
  <span class="printonly">(http://www.aroma-project.org/)</span>
  ::
  © Henrik Bengtsson
  ::
  Powered by <a href="http://cran.r-project.org/package=R.rsp">RSP</a>

  <span style="float: right;">
  <a href="http://validator.w3.org/check?st=1;outline=1;verbose=1;group=1&amp;uri=http://www.aroma-project.org/publications/" aria-label="Validate HTML" id="validate-page" target="_blank"><span class="glyphicon glyphicon-check" title="Validate HTML" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Validate HTML</span></a>
  &nbsp;
  <a href="https://github.com/AromaProject/aroma.project.org-website/tree/master/content/publications/index.md.rsp" id="edit"
  target="_blank"><span class="glyphicon glyphicon-edit" title="Edit page" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Edit page</span></a>
  &nbsp;
  <a href="http://v1.aroma-project.org/publications/" id="old-page"
  target="_blank"><span class="glyphicon glyphicon-time" title="Go to old page" aria-hidden="true"></span><span class="sr-only sr-only-focusable">Go to old page</span></a>
  </span>
</p>

  </div>
</div>

<div class="device-xs visible-xs"></div>
<div class="device-sm visible-sm"></div>
<div class="device-md visible-md"></div>
<div class="device-lg visible-lg"></div>

<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
<script src="../assets/js/post-jquery.js"></script>
<script src="http://netdna.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
<script src="../assets/js/post-bootstrap.js"></script>
  <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
    (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
    m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','//www.google-analytics.com/analytics.js','ga');
  
    ga('create', 'UA-11999401-1', 'auto');
    ga('send', 'pageview');
  </script>
</body>
</html>
